// react-native-agora/lib/module/internal/AgoraH265TranscoderInternal.js
import { createCheckers } from 'ts-interface-checker';
import { IH265TranscoderImpl } from '../impl/IAgoraH265TranscoderImpl';
import IAgoraH265TranscoderTI from '../ti/IAgoraH265Transcoder-ti';
const checkers = createCheckers(IAgoraH265TranscoderTI);
import { DeviceEventEmitter, EVENT_TYPE } from './IrisApiEngine';
export class H265TranscoderInternal extends IH265TranscoderImpl {
  static _h265_transcoder_observers = [];
  release() {
    H265TranscoderInternal._h265_transcoder_observers = [];
    this.removeAllListeners();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IH265Transc;
    if ((_checkers$IH265Transc = checkers.IH265TranscoderObserver) !== null && _checkers$IH265Transc !== void 0 && _checkers$IH265Transc.strictTest({
      [eventType]: undefined
    })) {
      if (H265TranscoderInternal._h265_transcoder_observers.length === 0) {
        this.registerTranscoderObserver({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== EVENT_TYPE.IAgoraH265Transcoder) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    DeviceEventEmitter.addListener(eventType, callback);
  }
  removeListener(eventType, listener) {
    DeviceEventEmitter.removeListener(eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    DeviceEventEmitter.removeAllListeners(eventType);
  }
  registerTranscoderObserver(observer) {
    if (!H265TranscoderInternal._h265_transcoder_observers.find(value => value === observer)) {
      H265TranscoderInternal._h265_transcoder_observers.push(observer);
    }
    return super.registerTranscoderObserver(observer);
  }
  unregisterTranscoderObserver(observer) {
    H265TranscoderInternal._h265_transcoder_observers = H265TranscoderInternal._h265_transcoder_observers.filter(value => value !== observer);
    return super.unregisterTranscoderObserver(observer);
  }
}
