// react-native-agora/lib/module/internal/MediaPlayerInternal.js
import { createCheckers } from 'ts-interface-checker';
import { ErrorCodeType } from '../AgoraBase';
import { RawAudioFrameOpModeType } from '../AgoraMediaBase';
import { IMediaPlayerImpl } from '../impl/IAgoraMediaPlayerImpl';
import AgoraMediaBaseTI from '../ti/AgoraMediaBase-ti';
import IAgoraMediaPlayerTI from '../ti/IAgoraMediaPlayer-ti';
import IAgoraMediaPlayerSourceTI from '../ti/IAgoraMediaPlayerSource-ti';
const checkers = createCheckers(AgoraMediaBaseTI, IAgoraMediaPlayerTI, IAgoraMediaPlayerSourceTI);
import { DeviceEventEmitter, EVENT_TYPE } from './IrisApiEngine';
export class MediaPlayerInternal extends IMediaPlayerImpl {
  static _source_observers = new Map();
  static _audio_frame_observers = new Map();
  static _video_frame_observers = new Map();
  static _audio_spectrum_observers = new Map();
  constructor(mediaPlayerId) {
    super();
    this._mediaPlayerId = mediaPlayerId;
  }
  release() {
    MediaPlayerInternal._source_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._audio_frame_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._video_frame_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._audio_spectrum_observers.delete(this._mediaPlayerId);
    this.removeAllListeners();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IMediaPlaye, _checkers$IMediaPlaye2, _checkers$IMediaPlaye3, _checkers$IAudioSpect;
    if ((_checkers$IMediaPlaye = checkers.IMediaPlayerSourceObserver) !== null && _checkers$IMediaPlaye !== void 0 && _checkers$IMediaPlaye.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$;
      if (((_MediaPlayerInternal$ = MediaPlayerInternal._source_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$ === void 0 ? void 0 : _MediaPlayerInternal$.length) === 0) {
        this.registerPlayerSourceObserver({});
      }
    }
    if ((_checkers$IMediaPlaye2 = checkers.IMediaPlayerAudioFrameObserver) !== null && _checkers$IMediaPlaye2 !== void 0 && _checkers$IMediaPlaye2.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$2;
      if (((_MediaPlayerInternal$2 = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$2 === void 0 ? void 0 : _MediaPlayerInternal$2.length) === 0) {
        this.registerAudioFrameObserver({});
      }
    }
    if ((_checkers$IMediaPlaye3 = checkers.IMediaPlayerVideoFrameObserver) !== null && _checkers$IMediaPlaye3 !== void 0 && _checkers$IMediaPlaye3.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$3;
      if (((_MediaPlayerInternal$3 = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$3 === void 0 ? void 0 : _MediaPlayerInternal$3.length) === 0) {
        this.registerVideoFrameObserver({});
      }
    }
    if ((_checkers$IAudioSpect = checkers.IAudioSpectrumObserver) !== null && _checkers$IAudioSpect !== void 0 && _checkers$IAudioSpect.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$4;
      if (((_MediaPlayerInternal$4 = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$4 === void 0 ? void 0 : _MediaPlayerInternal$4.length) === 0) {
        console.error('Please call `registerMediaPlayerAudioSpectrumObserver` before you want to receive event by `addListener`');
        return false;
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== EVENT_TYPE.IMediaPlayer) {
        return;
      }
      if (data.playerId !== this._mediaPlayerId) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    DeviceEventEmitter.addListener(eventType, callback);
  }
  removeListener(eventType, listener) {
    DeviceEventEmitter.removeListener(eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    DeviceEventEmitter.removeAllListeners(eventType);
  }
  getMediaPlayerId() {
    return this._mediaPlayerId;
  }
  registerPlayerSourceObserver(observer) {
    let observers = MediaPlayerInternal._source_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._source_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerPlayerSourceObserver(observer);
  }
  unregisterPlayerSourceObserver(observer) {
    let observers = MediaPlayerInternal._source_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -ErrorCodeType.ErrFailed;
    MediaPlayerInternal._source_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterPlayerSourceObserver(observer);
  }
  registerAudioFrameObserver(observer) {
    let mode = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : RawAudioFrameOpModeType.RawAudioFrameOpModeReadOnly;
    let observers = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._audio_frame_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerAudioFrameObserver(observer, mode);
  }
  unregisterAudioFrameObserver(observer) {
    let observers = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -ErrorCodeType.ErrFailed;
    MediaPlayerInternal._audio_frame_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterAudioFrameObserver(observer);
  }
  registerVideoFrameObserver(observer) {
    let observers = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._video_frame_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerVideoFrameObserver(observer);
  }
  unregisterVideoFrameObserver(observer) {
    let observers = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -ErrorCodeType.ErrFailed;
    MediaPlayerInternal._video_frame_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterVideoFrameObserver(observer);
  }
  registerMediaPlayerAudioSpectrumObserver(observer, intervalInMS) {
    let observers = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._audio_spectrum_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerMediaPlayerAudioSpectrumObserver(observer, intervalInMS);
  }
  unregisterMediaPlayerAudioSpectrumObserver(observer) {
    let observers = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -ErrorCodeType.ErrFailed;
    MediaPlayerInternal._audio_spectrum_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterMediaPlayerAudioSpectrumObserver(observer);
  }
}
