// react-native-agora/lib/module/impl/IAgoraMusicContentCenterImpl.js
import { IMediaPlayerImpl } from './IAgoraMediaPlayerImpl';

// @ts-ignore
export class MusicChartCollectionImpl {
  getCount() {
    const apiType = this.getApiTypeFromGetCount();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCount() {
    return 'MusicChartCollection_getCount';
  }
  get(index) {
    const apiType = this.getApiTypeFromGet(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGet(index) {
    return 'MusicChartCollection_get_46f8ab7';
  }
}

// @ts-ignore
export class MusicCollectionImpl {
  getCount() {
    const apiType = this.getApiTypeFromGetCount();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCount() {
    return 'MusicCollection_getCount';
  }
  getTotal() {
    const apiType = this.getApiTypeFromGetTotal();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetTotal() {
    return 'MusicCollection_getTotal';
  }
  getPage() {
    const apiType = this.getApiTypeFromGetPage();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetPage() {
    return 'MusicCollection_getPage';
  }
  getPageSize() {
    const apiType = this.getApiTypeFromGetPageSize();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetPageSize() {
    return 'MusicCollection_getPageSize';
  }
  getMusic(index) {
    const apiType = this.getApiTypeFromGetMusic(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMusic(index) {
    return 'MusicCollection_getMusic_8fcdcef';
  }
}
export function processIMusicContentCenterEventHandler(handler, event, jsonParams) {
  switch (event) {
    case 'onMusicChartsResult':
      if (handler.onMusicChartsResult !== undefined) {
        handler.onMusicChartsResult(jsonParams.requestId, jsonParams.result, jsonParams.reason);
      }
      break;
    case 'onMusicCollectionResult':
      if (handler.onMusicCollectionResult !== undefined) {
        handler.onMusicCollectionResult(jsonParams.requestId, jsonParams.result, jsonParams.reason);
      }
      break;
    case 'onLyricResult':
      if (handler.onLyricResult !== undefined) {
        handler.onLyricResult(jsonParams.requestId, jsonParams.songCode, jsonParams.lyricUrl, jsonParams.reason);
      }
      break;
    case 'onSongSimpleInfoResult':
      if (handler.onSongSimpleInfoResult !== undefined) {
        handler.onSongSimpleInfoResult(jsonParams.requestId, jsonParams.songCode, jsonParams.simpleInfo, jsonParams.reason);
      }
      break;
    case 'onPreLoadEvent':
      if (handler.onPreLoadEvent !== undefined) {
        handler.onPreLoadEvent(jsonParams.requestId, jsonParams.songCode, jsonParams.percent, jsonParams.lyricUrl, jsonParams.state, jsonParams.reason);
      }
      break;
  }
}

// @ts-ignore
export class IMusicPlayerImpl extends IMediaPlayerImpl {
  setPlayMode(mode) {
    const apiType = this.getApiTypeFromSetPlayMode(mode);
    const jsonParams = {
      mode: mode,
      toJSON: () => {
        return {
          mode: mode
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlayMode(mode) {
    return 'MusicPlayer_setPlayMode_748bee0';
  }
  openWithSongCode(songCode) {
    let startPos = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    const apiType = this.getApiTypeFromOpenWithSongCode(songCode, startPos);
    const jsonParams = {
      songCode: songCode,
      startPos: startPos,
      toJSON: () => {
        return {
          songCode: songCode,
          startPos: startPos
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromOpenWithSongCode(songCode) {
    let startPos = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'MusicPlayer_open_303b92e';
  }
}

// @ts-ignore
export class IMusicContentCenterImpl {
  initialize(configuration) {
    const apiType = this.getApiTypeFromInitialize(configuration);
    const jsonParams = {
      configuration: configuration,
      toJSON: () => {
        return {
          configuration: configuration
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromInitialize(configuration) {
    return 'MusicContentCenter_initialize_df70304';
  }
  renewToken(token) {
    const apiType = this.getApiTypeFromRenewToken(token);
    const jsonParams = {
      token: token,
      toJSON: () => {
        return {
          token: token
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRenewToken(token) {
    return 'MusicContentCenter_renewToken_3a2037f';
  }
  release() {
    const apiType = this.getApiTypeFromRelease();
    const jsonParams = {};
    callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromRelease() {
    return 'MusicContentCenter_release';
  }
  registerEventHandler(eventHandler) {
    const apiType = this.getApiTypeFromRegisterEventHandler(eventHandler);
    const jsonParams = {
      eventHandler: eventHandler,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterEventHandler(eventHandler) {
    return 'MusicContentCenter_registerEventHandler_ae49451';
  }
  unregisterEventHandler() {
    const apiType = this.getApiTypeFromUnregisterEventHandler();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterEventHandler() {
    return 'MusicContentCenter_unregisterEventHandler';
  }
  createMusicPlayer() {
    const apiType = this.getApiTypeFromCreateMusicPlayer();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateMusicPlayer() {
    return 'MusicContentCenter_createMusicPlayer';
  }
  destroyMusicPlayer(musicPlayer) {
    const apiType = this.getApiTypeFromDestroyMusicPlayer(musicPlayer);
    const jsonParams = {
      music_player: musicPlayer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyMusicPlayer(musicPlayer) {
    return 'MusicContentCenter_destroyMusicPlayer_876d086';
  }
  getMusicCharts() {
    const apiType = this.getApiTypeFromGetMusicCharts();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromGetMusicCharts() {
    return 'MusicContentCenter_getMusicCharts_66d4ecd';
  }
  getMusicCollectionByMusicChartId(musicChartId, page, pageSize, jsonOption) {
    const apiType = this.getApiTypeFromGetMusicCollectionByMusicChartId(musicChartId, page, pageSize, jsonOption);
    const jsonParams = {
      musicChartId: musicChartId,
      page: page,
      pageSize: pageSize,
      jsonOption: jsonOption,
      toJSON: () => {
        return {
          musicChartId: musicChartId,
          page: page,
          pageSize: pageSize,
          jsonOption: jsonOption
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromGetMusicCollectionByMusicChartId(musicChartId, page, pageSize, jsonOption) {
    return 'MusicContentCenter_getMusicCollectionByMusicChartId_8cd0b4d';
  }
  searchMusic(keyWord, page, pageSize, jsonOption) {
    const apiType = this.getApiTypeFromSearchMusic(keyWord, page, pageSize, jsonOption);
    const jsonParams = {
      keyWord: keyWord,
      page: page,
      pageSize: pageSize,
      jsonOption: jsonOption,
      toJSON: () => {
        return {
          keyWord: keyWord,
          page: page,
          pageSize: pageSize,
          jsonOption: jsonOption
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromSearchMusic(keyWord, page, pageSize, jsonOption) {
    return 'MusicContentCenter_searchMusic_3f8cf09';
  }
  preload(songCode) {
    const apiType = this.getApiTypeFromPreload(songCode);
    const jsonParams = {
      songCode: songCode,
      toJSON: () => {
        return {
          songCode: songCode
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromPreload(songCode) {
    return 'MusicContentCenter_preload_d3baeab';
  }
  removeCache(songCode) {
    const apiType = this.getApiTypeFromRemoveCache(songCode);
    const jsonParams = {
      songCode: songCode,
      toJSON: () => {
        return {
          songCode: songCode
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRemoveCache(songCode) {
    return 'MusicContentCenter_removeCache_f631116';
  }
  getCaches() {
    const apiType = this.getApiTypeFromGetCaches();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const cacheInfo = jsonResults.cacheInfo;
    const cacheInfoSize = jsonResults.cacheInfoSize;
    return {
      cacheInfo,
      cacheInfoSize
    };
  }
  getApiTypeFromGetCaches() {
    return 'MusicContentCenter_getCaches_c4f9978';
  }
  isPreloaded(songCode) {
    const apiType = this.getApiTypeFromIsPreloaded(songCode);
    const jsonParams = {
      songCode: songCode,
      toJSON: () => {
        return {
          songCode: songCode
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsPreloaded(songCode) {
    return 'MusicContentCenter_isPreloaded_f631116';
  }
  getLyric(songCode) {
    let lyricType = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    const apiType = this.getApiTypeFromGetLyric(songCode, lyricType);
    const jsonParams = {
      songCode: songCode,
      lyricType: lyricType,
      toJSON: () => {
        return {
          songCode: songCode,
          lyricType: lyricType
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromGetLyric(songCode) {
    let lyricType = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'MusicContentCenter_getLyric_5ab5efd';
  }
  getSongSimpleInfo(songCode) {
    const apiType = this.getApiTypeFromGetSongSimpleInfo(songCode);
    const jsonParams = {
      songCode: songCode,
      toJSON: () => {
        return {
          songCode: songCode
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromGetSongSimpleInfo(songCode) {
    return 'MusicContentCenter_getSongSimpleInfo_d3baeab';
  }
  getInternalSongCode(songCode, jsonOption) {
    const apiType = this.getApiTypeFromGetInternalSongCode(songCode, jsonOption);
    const jsonParams = {
      songCode: songCode,
      jsonOption: jsonOption,
      toJSON: () => {
        return {
          songCode: songCode,
          jsonOption: jsonOption
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const internalSongCode = jsonResults.internalSongCode;
    return internalSongCode;
  }
  getApiTypeFromGetInternalSongCode(songCode, jsonOption) {
    return 'MusicContentCenter_getInternalSongCode_3a3d1e7';
  }
}
import { callIrisApi } from '../internal/IrisApiEngine';
