// react-native-agora/lib/commonjs/impl/AgoraMediaBaseImpl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IVideoFrameMetaInfoImpl = void 0;
exports.processIAudioFrameObserver = processIAudioFrameObserver;
exports.processIAudioFrameObserverBase = processIAudioFrameObserverBase;
exports.processIAudioPcmFrameSink = processIAudioPcmFrameSink;
exports.processIAudioSpectrumObserver = processIAudioSpectrumObserver;
exports.processIFaceInfoObserver = processIFaceInfoObserver;
exports.processIMediaRecorderObserver = processIMediaRecorderObserver;
exports.processIVideoEncodedFrameObserver = processIVideoEncodedFrameObserver;
exports.processIVideoFrameObserver = processIVideoFrameObserver;
var _IrisApiEngine = require("../internal/IrisApiEngine");
// @ts-ignore
class IVideoFrameMetaInfoImpl {
  getMetaInfoStr(key) {
    const apiType = this.getApiTypeFromGetMetaInfoStr(key);
    const jsonParams = {
      key: key,
      toJSON: () => {
        return {
          key: key
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMetaInfoStr(key) {
    return 'VideoFrameMetaInfo_getMetaInfoStr_c81192f';
  }
}
exports.IVideoFrameMetaInfoImpl = IVideoFrameMetaInfoImpl;
function processIAudioPcmFrameSink(handler, event, jsonParams) {
  switch (event) {
    case 'onFrame':
      if (handler.onFrame !== undefined) {
        handler.onFrame(jsonParams.frame);
      }
      break;
  }
}
function processIAudioFrameObserverBase(handler, event, jsonParams) {
  switch (event) {
    case 'onRecordAudioFrame':
      if (handler.onRecordAudioFrame !== undefined) {
        handler.onRecordAudioFrame(jsonParams.channelId, jsonParams.audioFrame);
      }
      break;
    case 'onPlaybackAudioFrame':
      if (handler.onPlaybackAudioFrame !== undefined) {
        handler.onPlaybackAudioFrame(jsonParams.channelId, jsonParams.audioFrame);
      }
      break;
    case 'onMixedAudioFrame':
      if (handler.onMixedAudioFrame !== undefined) {
        handler.onMixedAudioFrame(jsonParams.channelId, jsonParams.audioFrame);
      }
      break;
    case 'onEarMonitoringAudioFrame':
      if (handler.onEarMonitoringAudioFrame !== undefined) {
        handler.onEarMonitoringAudioFrame(jsonParams.audioFrame);
      }
      break;
  }
}
function processIAudioFrameObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onPlaybackAudioFrameBeforeMixing':
      if (handler.onPlaybackAudioFrameBeforeMixing !== undefined) {
        handler.onPlaybackAudioFrameBeforeMixing(jsonParams.channelId, jsonParams.uid, jsonParams.audioFrame);
      }
      break;
  }
}
function processIAudioSpectrumObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onLocalAudioSpectrum':
      if (handler.onLocalAudioSpectrum !== undefined) {
        handler.onLocalAudioSpectrum(jsonParams.data);
      }
      break;
    case 'onRemoteAudioSpectrum':
      if (handler.onRemoteAudioSpectrum !== undefined) {
        handler.onRemoteAudioSpectrum(jsonParams.spectrums, jsonParams.spectrumNumber);
      }
      break;
  }
}
function processIVideoEncodedFrameObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onEncodedVideoFrameReceived':
      if (handler.onEncodedVideoFrameReceived !== undefined) {
        handler.onEncodedVideoFrameReceived(jsonParams.uid, jsonParams.imageBuffer, jsonParams.length, jsonParams.videoEncodedFrameInfo);
      }
      break;
  }
}
function processIVideoFrameObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onCaptureVideoFrame':
      if (handler.onCaptureVideoFrame !== undefined) {
        handler.onCaptureVideoFrame(jsonParams.sourceType, jsonParams.videoFrame);
      }
      break;
    case 'onPreEncodeVideoFrame':
      if (handler.onPreEncodeVideoFrame !== undefined) {
        handler.onPreEncodeVideoFrame(jsonParams.sourceType, jsonParams.videoFrame);
      }
      break;
    case 'onMediaPlayerVideoFrame':
      if (handler.onMediaPlayerVideoFrame !== undefined) {
        handler.onMediaPlayerVideoFrame(jsonParams.videoFrame, jsonParams.mediaPlayerId);
      }
      break;
    case 'onRenderVideoFrame':
      if (handler.onRenderVideoFrame !== undefined) {
        handler.onRenderVideoFrame(jsonParams.channelId, jsonParams.remoteUid, jsonParams.videoFrame);
      }
      break;
    case 'onTranscodedVideoFrame':
      if (handler.onTranscodedVideoFrame !== undefined) {
        handler.onTranscodedVideoFrame(jsonParams.videoFrame);
      }
      break;
  }
}
function processIFaceInfoObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onFaceInfo':
      if (handler.onFaceInfo !== undefined) {
        handler.onFaceInfo(jsonParams.outFaceInfo);
      }
      break;
  }
}
function processIMediaRecorderObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onRecorderStateChanged':
      if (handler.onRecorderStateChanged !== undefined) {
        handler.onRecorderStateChanged(jsonParams.channelId, jsonParams.uid, jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onRecorderInfoUpdated':
      if (handler.onRecorderInfoUpdated !== undefined) {
        handler.onRecorderInfoUpdated(jsonParams.channelId, jsonParams.uid, jsonParams.info);
      }
      break;
  }
}
