// react-native-agora/lib/commonjs/impl/IAgoraH265TranscoderImpl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IH265TranscoderImpl = void 0;
exports.processIH265TranscoderObserver = processIH265TranscoderObserver;
var _IrisApiEngine = require("../internal/IrisApiEngine");
function processIH265TranscoderObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onEnableTranscode':
      if (handler.onEnableTranscode !== undefined) {
        handler.onEnableTranscode(jsonParams.result);
      }
      break;
    case 'onQueryChannel':
      if (handler.onQueryChannel !== undefined) {
        handler.onQueryChannel(jsonParams.result, jsonParams.originChannel, jsonParams.transcodeChannel);
      }
      break;
    case 'onTriggerTranscode':
      if (handler.onTriggerTranscode !== undefined) {
        handler.onTriggerTranscode(jsonParams.result);
      }
      break;
  }
}

// @ts-ignore
class IH265TranscoderImpl {
  enableTranscode(token, channel, uid) {
    const apiType = this.getApiTypeFromEnableTranscode(token, channel, uid);
    const jsonParams = {
      token: token,
      channel: channel,
      uid: uid,
      toJSON: () => {
        return {
          token: token,
          channel: channel,
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableTranscode(token, channel, uid) {
    return 'H265Transcoder_enableTranscode_a0779eb';
  }
  queryChannel(token, channel, uid) {
    const apiType = this.getApiTypeFromQueryChannel(token, channel, uid);
    const jsonParams = {
      token: token,
      channel: channel,
      uid: uid,
      toJSON: () => {
        return {
          token: token,
          channel: channel,
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromQueryChannel(token, channel, uid) {
    return 'H265Transcoder_queryChannel_a0779eb';
  }
  triggerTranscode(token, channel, uid) {
    const apiType = this.getApiTypeFromTriggerTranscode(token, channel, uid);
    const jsonParams = {
      token: token,
      channel: channel,
      uid: uid,
      toJSON: () => {
        return {
          token: token,
          channel: channel,
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromTriggerTranscode(token, channel, uid) {
    return 'H265Transcoder_triggerTranscode_a0779eb';
  }
  registerTranscoderObserver(observer) {
    const apiType = this.getApiTypeFromRegisterTranscoderObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterTranscoderObserver(observer) {
    return 'H265Transcoder_registerTranscoderObserver_e1ee996';
  }
  unregisterTranscoderObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterTranscoderObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterTranscoderObserver(observer) {
    return 'H265Transcoder_unregisterTranscoderObserver_e1ee996';
  }
}
exports.IH265TranscoderImpl = IH265TranscoderImpl;
