// react-native-agora/lib/commonjs/impl/IAgoraRtcEngineImpl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IVideoDeviceManagerImpl = exports.IRtcEngineImpl = void 0;
exports.processIDirectCdnStreamingEventHandler = processIDirectCdnStreamingEventHandler;
exports.processIMetadataObserver = processIMetadataObserver;
exports.processIRtcEngineEventHandler = processIRtcEngineEventHandler;
var _AgoraBase = require("../AgoraBase");
var _AgoraMediaBase = require("../AgoraMediaBase");
var _IrisApiEngine = require("../internal/IrisApiEngine");
function processIRtcEngineEventHandler(handler, event, jsonParams) {
  switch (event) {
    case 'onJoinChannelSuccess':
      if (handler.onJoinChannelSuccess !== undefined) {
        handler.onJoinChannelSuccess(jsonParams.connection, jsonParams.elapsed);
      }
      break;
    case 'onRejoinChannelSuccess':
      if (handler.onRejoinChannelSuccess !== undefined) {
        handler.onRejoinChannelSuccess(jsonParams.connection, jsonParams.elapsed);
      }
      break;
    case 'onProxyConnected':
      if (handler.onProxyConnected !== undefined) {
        handler.onProxyConnected(jsonParams.channel, jsonParams.uid, jsonParams.proxyType, jsonParams.localProxyIp, jsonParams.elapsed);
      }
      break;
    case 'onError':
      if (handler.onError !== undefined) {
        handler.onError(jsonParams.err, jsonParams.msg);
      }
      break;
    case 'onAudioQuality':
      if (handler.onAudioQuality !== undefined) {
        handler.onAudioQuality(jsonParams.connection, jsonParams.remoteUid, jsonParams.quality, jsonParams.delay, jsonParams.lost);
      }
      break;
    case 'onLastmileProbeResult':
      if (handler.onLastmileProbeResult !== undefined) {
        handler.onLastmileProbeResult(jsonParams.result);
      }
      break;
    case 'onAudioVolumeIndication':
      if (handler.onAudioVolumeIndication !== undefined) {
        handler.onAudioVolumeIndication(jsonParams.connection, jsonParams.speakers, jsonParams.speakerNumber, jsonParams.totalVolume);
      }
      break;
    case 'onLeaveChannel':
      if (handler.onLeaveChannel !== undefined) {
        handler.onLeaveChannel(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onRtcStats':
      if (handler.onRtcStats !== undefined) {
        handler.onRtcStats(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onAudioDeviceStateChanged':
      if (handler.onAudioDeviceStateChanged !== undefined) {
        handler.onAudioDeviceStateChanged(jsonParams.deviceId, jsonParams.deviceType, jsonParams.deviceState);
      }
      break;
    case 'onAudioMixingPositionChanged':
      if (handler.onAudioMixingPositionChanged !== undefined) {
        handler.onAudioMixingPositionChanged(jsonParams.position);
      }
      break;
    case 'onAudioMixingFinished':
      if (handler.onAudioMixingFinished !== undefined) {
        handler.onAudioMixingFinished();
      }
      break;
    case 'onAudioEffectFinished':
      if (handler.onAudioEffectFinished !== undefined) {
        handler.onAudioEffectFinished(jsonParams.soundId);
      }
      break;
    case 'onVideoDeviceStateChanged':
      if (handler.onVideoDeviceStateChanged !== undefined) {
        handler.onVideoDeviceStateChanged(jsonParams.deviceId, jsonParams.deviceType, jsonParams.deviceState);
      }
      break;
    case 'onNetworkQuality':
      if (handler.onNetworkQuality !== undefined) {
        handler.onNetworkQuality(jsonParams.connection, jsonParams.remoteUid, jsonParams.txQuality, jsonParams.rxQuality);
      }
      break;
    case 'onIntraRequestReceived':
      if (handler.onIntraRequestReceived !== undefined) {
        handler.onIntraRequestReceived(jsonParams.connection);
      }
      break;
    case 'onUplinkNetworkInfoUpdated':
      if (handler.onUplinkNetworkInfoUpdated !== undefined) {
        handler.onUplinkNetworkInfoUpdated(jsonParams.info);
      }
      break;
    case 'onDownlinkNetworkInfoUpdated':
      if (handler.onDownlinkNetworkInfoUpdated !== undefined) {
        handler.onDownlinkNetworkInfoUpdated(jsonParams.info);
      }
      break;
    case 'onLastmileQuality':
      if (handler.onLastmileQuality !== undefined) {
        handler.onLastmileQuality(jsonParams.quality);
      }
      break;
    case 'onFirstLocalVideoFrame':
      if (handler.onFirstLocalVideoFrame !== undefined) {
        handler.onFirstLocalVideoFrame(jsonParams.source, jsonParams.width, jsonParams.height, jsonParams.elapsed);
      }
      break;
    case 'onFirstLocalVideoFramePublished':
      if (handler.onFirstLocalVideoFramePublished !== undefined) {
        handler.onFirstLocalVideoFramePublished(jsonParams.connection, jsonParams.elapsed);
      }
      break;
    case 'onFirstRemoteVideoDecoded':
      if (handler.onFirstRemoteVideoDecoded !== undefined) {
        handler.onFirstRemoteVideoDecoded(jsonParams.connection, jsonParams.remoteUid, jsonParams.width, jsonParams.height, jsonParams.elapsed);
      }
      break;
    case 'onVideoSizeChanged':
      if (handler.onVideoSizeChanged !== undefined) {
        handler.onVideoSizeChanged(jsonParams.connection, jsonParams.sourceType, jsonParams.uid, jsonParams.width, jsonParams.height, jsonParams.rotation);
      }
      break;
    case 'onLocalVideoStateChanged':
      if (handler.onLocalVideoStateChanged !== undefined) {
        handler.onLocalVideoStateChanged(jsonParams.source, jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onRemoteVideoStateChanged':
      if (handler.onRemoteVideoStateChanged !== undefined) {
        handler.onRemoteVideoStateChanged(jsonParams.connection, jsonParams.remoteUid, jsonParams.state, jsonParams.reason, jsonParams.elapsed);
      }
      break;
    case 'onFirstRemoteVideoFrame':
      if (handler.onFirstRemoteVideoFrame !== undefined) {
        handler.onFirstRemoteVideoFrame(jsonParams.connection, jsonParams.remoteUid, jsonParams.width, jsonParams.height, jsonParams.elapsed);
      }
      break;
    case 'onUserJoined':
      if (handler.onUserJoined !== undefined) {
        handler.onUserJoined(jsonParams.connection, jsonParams.remoteUid, jsonParams.elapsed);
      }
      break;
    case 'onUserOffline':
      if (handler.onUserOffline !== undefined) {
        handler.onUserOffline(jsonParams.connection, jsonParams.remoteUid, jsonParams.reason);
      }
      break;
    case 'onUserMuteAudio':
      if (handler.onUserMuteAudio !== undefined) {
        handler.onUserMuteAudio(jsonParams.connection, jsonParams.remoteUid, jsonParams.muted);
      }
      break;
    case 'onUserMuteVideo':
      if (handler.onUserMuteVideo !== undefined) {
        handler.onUserMuteVideo(jsonParams.connection, jsonParams.remoteUid, jsonParams.muted);
      }
      break;
    case 'onUserEnableVideo':
      if (handler.onUserEnableVideo !== undefined) {
        handler.onUserEnableVideo(jsonParams.connection, jsonParams.remoteUid, jsonParams.enabled);
      }
      break;
    case 'onUserStateChanged':
      if (handler.onUserStateChanged !== undefined) {
        handler.onUserStateChanged(jsonParams.connection, jsonParams.remoteUid, jsonParams.state);
      }
      break;
    case 'onUserEnableLocalVideo':
      if (handler.onUserEnableLocalVideo !== undefined) {
        handler.onUserEnableLocalVideo(jsonParams.connection, jsonParams.remoteUid, jsonParams.enabled);
      }
      break;
    case 'onRemoteAudioStats':
      if (handler.onRemoteAudioStats !== undefined) {
        handler.onRemoteAudioStats(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onLocalAudioStats':
      if (handler.onLocalAudioStats !== undefined) {
        handler.onLocalAudioStats(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onLocalVideoStats':
      if (handler.onLocalVideoStats !== undefined) {
        handler.onLocalVideoStats(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onRemoteVideoStats':
      if (handler.onRemoteVideoStats !== undefined) {
        handler.onRemoteVideoStats(jsonParams.connection, jsonParams.stats);
      }
      break;
    case 'onCameraReady':
      if (handler.onCameraReady !== undefined) {
        handler.onCameraReady();
      }
      break;
    case 'onCameraFocusAreaChanged':
      if (handler.onCameraFocusAreaChanged !== undefined) {
        handler.onCameraFocusAreaChanged(jsonParams.x, jsonParams.y, jsonParams.width, jsonParams.height);
      }
      break;
    case 'onCameraExposureAreaChanged':
      if (handler.onCameraExposureAreaChanged !== undefined) {
        handler.onCameraExposureAreaChanged(jsonParams.x, jsonParams.y, jsonParams.width, jsonParams.height);
      }
      break;
    case 'onFacePositionChanged':
      if (handler.onFacePositionChanged !== undefined) {
        handler.onFacePositionChanged(jsonParams.imageWidth, jsonParams.imageHeight, jsonParams.vecRectangle, jsonParams.vecDistance, jsonParams.numFaces);
      }
      break;
    case 'onVideoStopped':
      if (handler.onVideoStopped !== undefined) {
        handler.onVideoStopped();
      }
      break;
    case 'onAudioMixingStateChanged':
      if (handler.onAudioMixingStateChanged !== undefined) {
        handler.onAudioMixingStateChanged(jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onRhythmPlayerStateChanged':
      if (handler.onRhythmPlayerStateChanged !== undefined) {
        handler.onRhythmPlayerStateChanged(jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onConnectionLost':
      if (handler.onConnectionLost !== undefined) {
        handler.onConnectionLost(jsonParams.connection);
      }
      break;
    case 'onConnectionInterrupted':
      if (handler.onConnectionInterrupted !== undefined) {
        handler.onConnectionInterrupted(jsonParams.connection);
      }
      break;
    case 'onConnectionBanned':
      if (handler.onConnectionBanned !== undefined) {
        handler.onConnectionBanned(jsonParams.connection);
      }
      break;
    case 'onStreamMessage':
      if (handler.onStreamMessage !== undefined) {
        handler.onStreamMessage(jsonParams.connection, jsonParams.remoteUid, jsonParams.streamId, jsonParams.data, jsonParams.length, jsonParams.sentTs);
      }
      break;
    case 'onStreamMessageError':
      if (handler.onStreamMessageError !== undefined) {
        handler.onStreamMessageError(jsonParams.connection, jsonParams.remoteUid, jsonParams.streamId, jsonParams.code, jsonParams.missed, jsonParams.cached);
      }
      break;
    case 'onRequestToken':
      if (handler.onRequestToken !== undefined) {
        handler.onRequestToken(jsonParams.connection);
      }
      break;
    case 'onTokenPrivilegeWillExpire':
      if (handler.onTokenPrivilegeWillExpire !== undefined) {
        handler.onTokenPrivilegeWillExpire(jsonParams.connection, jsonParams.token);
      }
      break;
    case 'onLicenseValidationFailure':
      if (handler.onLicenseValidationFailure !== undefined) {
        handler.onLicenseValidationFailure(jsonParams.connection, jsonParams.reason);
      }
      break;
    case 'onFirstLocalAudioFramePublished':
      if (handler.onFirstLocalAudioFramePublished !== undefined) {
        handler.onFirstLocalAudioFramePublished(jsonParams.connection, jsonParams.elapsed);
      }
      break;
    case 'onFirstRemoteAudioDecoded':
      if (handler.onFirstRemoteAudioDecoded !== undefined) {
        handler.onFirstRemoteAudioDecoded(jsonParams.connection, jsonParams.uid, jsonParams.elapsed);
      }
      break;
    case 'onFirstRemoteAudioFrame':
      if (handler.onFirstRemoteAudioFrame !== undefined) {
        handler.onFirstRemoteAudioFrame(jsonParams.connection, jsonParams.userId, jsonParams.elapsed);
      }
      break;
    case 'onLocalAudioStateChanged':
      if (handler.onLocalAudioStateChanged !== undefined) {
        handler.onLocalAudioStateChanged(jsonParams.connection, jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onRemoteAudioStateChanged':
      if (handler.onRemoteAudioStateChanged !== undefined) {
        handler.onRemoteAudioStateChanged(jsonParams.connection, jsonParams.remoteUid, jsonParams.state, jsonParams.reason, jsonParams.elapsed);
      }
      break;
    case 'onActiveSpeaker':
      if (handler.onActiveSpeaker !== undefined) {
        handler.onActiveSpeaker(jsonParams.connection, jsonParams.uid);
      }
      break;
    case 'onContentInspectResult':
      if (handler.onContentInspectResult !== undefined) {
        handler.onContentInspectResult(jsonParams.result);
      }
      break;
    case 'onSnapshotTaken':
      if (handler.onSnapshotTaken !== undefined) {
        handler.onSnapshotTaken(jsonParams.connection, jsonParams.uid, jsonParams.filePath, jsonParams.width, jsonParams.height, jsonParams.errCode);
      }
      break;
    case 'onClientRoleChanged':
      if (handler.onClientRoleChanged !== undefined) {
        handler.onClientRoleChanged(jsonParams.connection, jsonParams.oldRole, jsonParams.newRole, jsonParams.newRoleOptions);
      }
      break;
    case 'onClientRoleChangeFailed':
      if (handler.onClientRoleChangeFailed !== undefined) {
        handler.onClientRoleChangeFailed(jsonParams.connection, jsonParams.reason, jsonParams.currentRole);
      }
      break;
    case 'onAudioDeviceVolumeChanged':
      if (handler.onAudioDeviceVolumeChanged !== undefined) {
        handler.onAudioDeviceVolumeChanged(jsonParams.deviceType, jsonParams.volume, jsonParams.muted);
      }
      break;
    case 'onRtmpStreamingStateChanged':
      if (handler.onRtmpStreamingStateChanged !== undefined) {
        handler.onRtmpStreamingStateChanged(jsonParams.url, jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onRtmpStreamingEvent':
      if (handler.onRtmpStreamingEvent !== undefined) {
        handler.onRtmpStreamingEvent(jsonParams.url, jsonParams.eventCode);
      }
      break;
    case 'onTranscodingUpdated':
      if (handler.onTranscodingUpdated !== undefined) {
        handler.onTranscodingUpdated();
      }
      break;
    case 'onAudioRoutingChanged':
      if (handler.onAudioRoutingChanged !== undefined) {
        handler.onAudioRoutingChanged(jsonParams.routing);
      }
      break;
    case 'onChannelMediaRelayStateChanged':
      if (handler.onChannelMediaRelayStateChanged !== undefined) {
        handler.onChannelMediaRelayStateChanged(jsonParams.state, jsonParams.code);
      }
      break;
    case 'onLocalPublishFallbackToAudioOnly':
      if (handler.onLocalPublishFallbackToAudioOnly !== undefined) {
        handler.onLocalPublishFallbackToAudioOnly(jsonParams.isFallbackOrRecover);
      }
      break;
    case 'onRemoteSubscribeFallbackToAudioOnly':
      if (handler.onRemoteSubscribeFallbackToAudioOnly !== undefined) {
        handler.onRemoteSubscribeFallbackToAudioOnly(jsonParams.uid, jsonParams.isFallbackOrRecover);
      }
      break;
    case 'onRemoteAudioTransportStats':
      if (handler.onRemoteAudioTransportStats !== undefined) {
        handler.onRemoteAudioTransportStats(jsonParams.connection, jsonParams.remoteUid, jsonParams.delay, jsonParams.lost, jsonParams.rxKBitRate);
      }
      break;
    case 'onRemoteVideoTransportStats':
      if (handler.onRemoteVideoTransportStats !== undefined) {
        handler.onRemoteVideoTransportStats(jsonParams.connection, jsonParams.remoteUid, jsonParams.delay, jsonParams.lost, jsonParams.rxKBitRate);
      }
      break;
    case 'onConnectionStateChanged':
      if (handler.onConnectionStateChanged !== undefined) {
        handler.onConnectionStateChanged(jsonParams.connection, jsonParams.state, jsonParams.reason);
      }
      break;
    case 'onWlAccMessage':
      if (handler.onWlAccMessage !== undefined) {
        handler.onWlAccMessage(jsonParams.connection, jsonParams.reason, jsonParams.action, jsonParams.wlAccMsg);
      }
      break;
    case 'onWlAccStats':
      if (handler.onWlAccStats !== undefined) {
        handler.onWlAccStats(jsonParams.connection, jsonParams.currentStats, jsonParams.averageStats);
      }
      break;
    case 'onNetworkTypeChanged':
      if (handler.onNetworkTypeChanged !== undefined) {
        handler.onNetworkTypeChanged(jsonParams.connection, jsonParams.type);
      }
      break;
    case 'onEncryptionError':
      if (handler.onEncryptionError !== undefined) {
        handler.onEncryptionError(jsonParams.connection, jsonParams.errorType);
      }
      break;
    case 'onPermissionError':
      if (handler.onPermissionError !== undefined) {
        handler.onPermissionError(jsonParams.permissionType);
      }
      break;
    case 'onLocalUserRegistered':
      if (handler.onLocalUserRegistered !== undefined) {
        handler.onLocalUserRegistered(jsonParams.uid, jsonParams.userAccount);
      }
      break;
    case 'onUserInfoUpdated':
      if (handler.onUserInfoUpdated !== undefined) {
        handler.onUserInfoUpdated(jsonParams.uid, jsonParams.info);
      }
      break;
    case 'onUserAccountUpdated':
      if (handler.onUserAccountUpdated !== undefined) {
        handler.onUserAccountUpdated(jsonParams.connection, jsonParams.remoteUid, jsonParams.remoteUserAccount);
      }
      break;
    case 'onVideoRenderingTracingResult':
      if (handler.onVideoRenderingTracingResult !== undefined) {
        handler.onVideoRenderingTracingResult(jsonParams.connection, jsonParams.uid, jsonParams.currentEvent, jsonParams.tracingInfo);
      }
      break;
    case 'onLocalVideoTranscoderError':
      if (handler.onLocalVideoTranscoderError !== undefined) {
        handler.onLocalVideoTranscoderError(jsonParams.stream, jsonParams.error);
      }
      break;
    case 'onUploadLogResult':
      if (handler.onUploadLogResult !== undefined) {
        handler.onUploadLogResult(jsonParams.connection, jsonParams.requestId, jsonParams.success, jsonParams.reason);
      }
      break;
    case 'onAudioSubscribeStateChanged':
      if (handler.onAudioSubscribeStateChanged !== undefined) {
        handler.onAudioSubscribeStateChanged(jsonParams.channel, jsonParams.uid, jsonParams.oldState, jsonParams.newState, jsonParams.elapseSinceLastState);
      }
      break;
    case 'onVideoSubscribeStateChanged':
      if (handler.onVideoSubscribeStateChanged !== undefined) {
        handler.onVideoSubscribeStateChanged(jsonParams.channel, jsonParams.uid, jsonParams.oldState, jsonParams.newState, jsonParams.elapseSinceLastState);
      }
      break;
    case 'onAudioPublishStateChanged':
      if (handler.onAudioPublishStateChanged !== undefined) {
        handler.onAudioPublishStateChanged(jsonParams.channel, jsonParams.oldState, jsonParams.newState, jsonParams.elapseSinceLastState);
      }
      break;
    case 'onVideoPublishStateChanged':
      if (handler.onVideoPublishStateChanged !== undefined) {
        handler.onVideoPublishStateChanged(jsonParams.source, jsonParams.channel, jsonParams.oldState, jsonParams.newState, jsonParams.elapseSinceLastState);
      }
      break;
    case 'onTranscodedStreamLayoutInfo':
      if (handler.onTranscodedStreamLayoutInfo !== undefined) {
        handler.onTranscodedStreamLayoutInfo(jsonParams.connection, jsonParams.uid, jsonParams.width, jsonParams.height, jsonParams.layoutCount, jsonParams.layoutlist);
      }
      break;
    case 'onAudioMetadataReceived':
      if (handler.onAudioMetadataReceived !== undefined) {
        handler.onAudioMetadataReceived(jsonParams.connection, jsonParams.uid, jsonParams.metadata, jsonParams.length);
      }
      break;
    case 'onExtensionEventWithContext':
      if (handler.onExtensionEventWithContext !== undefined) {
        handler.onExtensionEventWithContext(jsonParams.context, jsonParams.key, jsonParams.value);
      }
      break;
    case 'onExtensionStartedWithContext':
      if (handler.onExtensionStartedWithContext !== undefined) {
        handler.onExtensionStartedWithContext(jsonParams.context);
      }
      break;
    case 'onExtensionStoppedWithContext':
      if (handler.onExtensionStoppedWithContext !== undefined) {
        handler.onExtensionStoppedWithContext(jsonParams.context);
      }
      break;
    case 'onExtensionErrorWithContext':
      if (handler.onExtensionErrorWithContext !== undefined) {
        handler.onExtensionErrorWithContext(jsonParams.context, jsonParams.error, jsonParams.message);
      }
      break;
    case 'onSetRtmFlagResult':
      if (handler.onSetRtmFlagResult !== undefined) {
        handler.onSetRtmFlagResult(jsonParams.connection, jsonParams.code);
      }
      break;
  }
}

// @ts-ignore
class IVideoDeviceManagerImpl {
  enumerateVideoDevices() {
    const apiType = this.getApiTypeFromEnumerateVideoDevices();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnumerateVideoDevices() {
    return 'VideoDeviceManager_enumerateVideoDevices';
  }
  setDevice(deviceIdUTF8) {
    const apiType = this.getApiTypeFromSetDevice(deviceIdUTF8);
    const jsonParams = {
      deviceIdUTF8: deviceIdUTF8,
      toJSON: () => {
        return {
          deviceIdUTF8: deviceIdUTF8
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDevice(deviceIdUTF8) {
    return 'VideoDeviceManager_setDevice_4ad5f6e';
  }
  getDevice() {
    const apiType = this.getApiTypeFromGetDevice();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const deviceIdUTF8 = jsonResults.deviceIdUTF8;
    return deviceIdUTF8;
  }
  getApiTypeFromGetDevice() {
    return 'VideoDeviceManager_getDevice_73b9872';
  }
  numberOfCapabilities(deviceIdUTF8) {
    const apiType = this.getApiTypeFromNumberOfCapabilities(deviceIdUTF8);
    const jsonParams = {
      deviceIdUTF8: deviceIdUTF8,
      toJSON: () => {
        return {
          deviceIdUTF8: deviceIdUTF8
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromNumberOfCapabilities(deviceIdUTF8) {
    return 'VideoDeviceManager_numberOfCapabilities_3a2037f';
  }
  getCapability(deviceIdUTF8, deviceCapabilityNumber) {
    const apiType = this.getApiTypeFromGetCapability(deviceIdUTF8, deviceCapabilityNumber);
    const jsonParams = {
      deviceIdUTF8: deviceIdUTF8,
      deviceCapabilityNumber: deviceCapabilityNumber,
      toJSON: () => {
        return {
          deviceIdUTF8: deviceIdUTF8,
          deviceCapabilityNumber: deviceCapabilityNumber
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const capability = jsonResults.capability;
    return capability;
  }
  getApiTypeFromGetCapability(deviceIdUTF8, deviceCapabilityNumber) {
    return 'VideoDeviceManager_getCapability_ddeefdd';
  }
  startDeviceTest(hwnd) {
    const apiType = this.getApiTypeFromStartDeviceTest(hwnd);
    const jsonParams = {
      hwnd: hwnd,
      toJSON: () => {
        return {
          hwnd: hwnd
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartDeviceTest(hwnd) {
    return 'VideoDeviceManager_startDeviceTest_a55f55f';
  }
  stopDeviceTest() {
    const apiType = this.getApiTypeFromStopDeviceTest();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopDeviceTest() {
    return 'VideoDeviceManager_stopDeviceTest';
  }
  release() {
    const apiType = this.getApiTypeFromRelease();
    const jsonParams = {};
    _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromRelease() {
    return 'VideoDeviceManager_release';
  }
}
exports.IVideoDeviceManagerImpl = IVideoDeviceManagerImpl;
function processIMetadataObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onMetadataReceived':
      if (handler.onMetadataReceived !== undefined) {
        handler.onMetadataReceived(jsonParams.metadata);
      }
      break;
  }
}
function processIDirectCdnStreamingEventHandler(handler, event, jsonParams) {
  switch (event) {
    case 'onDirectCdnStreamingStateChanged':
      if (handler.onDirectCdnStreamingStateChanged !== undefined) {
        handler.onDirectCdnStreamingStateChanged(jsonParams.state, jsonParams.reason, jsonParams.message);
      }
      break;
    case 'onDirectCdnStreamingStats':
      if (handler.onDirectCdnStreamingStats !== undefined) {
        handler.onDirectCdnStreamingStats(jsonParams.stats);
      }
      break;
  }
}

// @ts-ignore
class IRtcEngineImpl {
  initialize(context) {
    const apiType = this.getApiTypeFromInitialize(context);
    const jsonParams = {
      context: context,
      toJSON: () => {
        return {
          context: context
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromInitialize(context) {
    return 'RtcEngine_initialize_0320339';
  }
  getVersion() {
    const apiType = this.getApiTypeFromGetVersion();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetVersion() {
    return 'RtcEngine_getVersion_915cb25';
  }
  getErrorDescription(code) {
    const apiType = this.getApiTypeFromGetErrorDescription(code);
    const jsonParams = {
      code: code,
      toJSON: () => {
        return {
          code: code
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetErrorDescription(code) {
    return 'RtcEngine_getErrorDescription_46f8ab7';
  }
  queryCodecCapability() {
    const apiType = this.getApiTypeFromQueryCodecCapability();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const codecInfo = jsonResults.codecInfo;
    const size = jsonResults.size;
    return {
      codecInfo,
      size
    };
  }
  getApiTypeFromQueryCodecCapability() {
    return 'RtcEngine_queryCodecCapability_ddf4f31';
  }
  queryDeviceScore() {
    const apiType = this.getApiTypeFromQueryDeviceScore();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromQueryDeviceScore() {
    return 'RtcEngine_queryDeviceScore';
  }
  preloadChannel(token, channelId, uid) {
    const apiType = this.getApiTypeFromPreloadChannel(token, channelId, uid);
    const jsonParams = {
      token: token,
      channelId: channelId,
      uid: uid,
      toJSON: () => {
        return {
          token: token,
          channelId: channelId,
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPreloadChannel(token, channelId, uid) {
    return 'RtcEngine_preloadChannel_a0779eb';
  }
  preloadChannelWithUserAccount(token, channelId, userAccount) {
    const apiType = this.getApiTypeFromPreloadChannelWithUserAccount(token, channelId, userAccount);
    const jsonParams = {
      token: token,
      channelId: channelId,
      userAccount: userAccount,
      toJSON: () => {
        return {
          token: token,
          channelId: channelId,
          userAccount: userAccount
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPreloadChannelWithUserAccount(token, channelId, userAccount) {
    return 'RtcEngine_preloadChannelWithUserAccount_0e4f59e';
  }
  updatePreloadChannelToken(token) {
    const apiType = this.getApiTypeFromUpdatePreloadChannelToken(token);
    const jsonParams = {
      token: token,
      toJSON: () => {
        return {
          token: token
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdatePreloadChannelToken(token) {
    return 'RtcEngine_updatePreloadChannelToken_3a2037f';
  }
  joinChannel(token, channelId, uid, options) {
    const apiType = this.getApiTypeFromJoinChannel(token, channelId, uid, options);
    const jsonParams = {
      token: token,
      channelId: channelId,
      uid: uid,
      options: options,
      toJSON: () => {
        return {
          token: token,
          channelId: channelId,
          uid: uid,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromJoinChannel(token, channelId, uid, options) {
    return 'RtcEngine_joinChannel_cdbb747';
  }
  updateChannelMediaOptions(options) {
    const apiType = this.getApiTypeFromUpdateChannelMediaOptions(options);
    const jsonParams = {
      options: options,
      toJSON: () => {
        return {
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateChannelMediaOptions(options) {
    return 'RtcEngine_updateChannelMediaOptions_7bfc1d7';
  }
  leaveChannel(options) {
    const apiType = this.getApiTypeFromLeaveChannel(options);
    const jsonParams = {
      options: options,
      toJSON: () => {
        return {
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromLeaveChannel(options) {
    return 'RtcEngine_leaveChannel_2c0e3aa';
  }
  renewToken(token) {
    const apiType = this.getApiTypeFromRenewToken(token);
    const jsonParams = {
      token: token,
      toJSON: () => {
        return {
          token: token
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRenewToken(token) {
    return 'RtcEngine_renewToken_3a2037f';
  }
  setChannelProfile(profile) {
    const apiType = this.getApiTypeFromSetChannelProfile(profile);
    const jsonParams = {
      profile: profile,
      toJSON: () => {
        return {
          profile: profile
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetChannelProfile(profile) {
    return 'RtcEngine_setChannelProfile_a78fa4f';
  }
  setClientRole(role, options) {
    const apiType = this.getApiTypeFromSetClientRole(role, options);
    const jsonParams = {
      role: role,
      options: options,
      toJSON: () => {
        return {
          role: role,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetClientRole(role, options) {
    return 'RtcEngine_setClientRole_b46cc48';
  }
  startEchoTest(config) {
    const apiType = this.getApiTypeFromStartEchoTest(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartEchoTest(config) {
    return 'RtcEngine_startEchoTest_16140d7';
  }
  stopEchoTest() {
    const apiType = this.getApiTypeFromStopEchoTest();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopEchoTest() {
    return 'RtcEngine_stopEchoTest';
  }
  enableMultiCamera(enabled, config) {
    const apiType = this.getApiTypeFromEnableMultiCamera(enabled, config);
    const jsonParams = {
      enabled: enabled,
      config: config,
      toJSON: () => {
        return {
          enabled: enabled,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableMultiCamera(enabled, config) {
    return 'RtcEngine_enableMultiCamera_bffe023';
  }
  enableVideo() {
    const apiType = this.getApiTypeFromEnableVideo();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableVideo() {
    return 'RtcEngine_enableVideo';
  }
  disableVideo() {
    const apiType = this.getApiTypeFromDisableVideo();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDisableVideo() {
    return 'RtcEngine_disableVideo';
  }
  startPreview() {
    let sourceType = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.VideoSourceType.VideoSourceCameraPrimary;
    const apiType = this.getApiTypeFromStartPreview(sourceType);
    const jsonParams = {
      sourceType: sourceType,
      toJSON: () => {
        return {
          sourceType: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartPreview() {
    let sourceType = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.VideoSourceType.VideoSourceCameraPrimary;
    return 'RtcEngine_startPreview_4fd718e';
  }
  stopPreview() {
    let sourceType = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.VideoSourceType.VideoSourceCameraPrimary;
    const apiType = this.getApiTypeFromStopPreview(sourceType);
    const jsonParams = {
      sourceType: sourceType,
      toJSON: () => {
        return {
          sourceType: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopPreview() {
    let sourceType = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.VideoSourceType.VideoSourceCameraPrimary;
    return 'RtcEngine_stopPreview_4fd718e';
  }
  startLastmileProbeTest(config) {
    const apiType = this.getApiTypeFromStartLastmileProbeTest(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartLastmileProbeTest(config) {
    return 'RtcEngine_startLastmileProbeTest_c4de423';
  }
  stopLastmileProbeTest() {
    const apiType = this.getApiTypeFromStopLastmileProbeTest();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopLastmileProbeTest() {
    return 'RtcEngine_stopLastmileProbeTest';
  }
  setVideoEncoderConfiguration(config) {
    const apiType = this.getApiTypeFromSetVideoEncoderConfiguration(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVideoEncoderConfiguration(config) {
    return 'RtcEngine_setVideoEncoderConfiguration_89677d8';
  }
  setBeautyEffectOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetBeautyEffectOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetBeautyEffectOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setBeautyEffectOptions_e7635d1';
  }
  setFaceShapeBeautyOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetFaceShapeBeautyOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetFaceShapeBeautyOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setFaceShapeBeautyOptions_a862ce7';
  }
  setFaceShapeAreaOptions(options) {
    let type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetFaceShapeAreaOptions(options, type);
    const jsonParams = {
      options: options,
      type: type,
      toJSON: () => {
        return {
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetFaceShapeAreaOptions(options) {
    let type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setFaceShapeAreaOptions_2e242a3';
  }
  getFaceShapeBeautyOptions() {
    let type = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromGetFaceShapeBeautyOptions(type);
    const jsonParams = {
      type: type,
      toJSON: () => {
        return {
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const options = jsonResults.options;
    return options;
  }
  getApiTypeFromGetFaceShapeBeautyOptions() {
    let type = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_getFaceShapeBeautyOptions_8382895';
  }
  getFaceShapeAreaOptions(shapeArea) {
    let type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromGetFaceShapeAreaOptions(shapeArea, type);
    const jsonParams = {
      shapeArea: shapeArea,
      type: type,
      toJSON: () => {
        return {
          shapeArea: shapeArea,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const options = jsonResults.options;
    return options;
  }
  getApiTypeFromGetFaceShapeAreaOptions(shapeArea) {
    let type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_getFaceShapeAreaOptions_0783e2c';
  }
  setFilterEffectOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetFilterEffectOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetFilterEffectOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setFilterEffectOptions_53b4be3';
  }
  setLowlightEnhanceOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetLowlightEnhanceOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLowlightEnhanceOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setLowlightEnhanceOptions_4f9f013';
  }
  setVideoDenoiserOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetVideoDenoiserOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVideoDenoiserOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setVideoDenoiserOptions_4e9fccc';
  }
  setColorEnhanceOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromSetColorEnhanceOptions(enabled, options, type);
    const jsonParams = {
      enabled: enabled,
      options: options,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          options: options,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetColorEnhanceOptions(enabled, options) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_setColorEnhanceOptions_ecae2b3';
  }
  enableVirtualBackground(enabled, backgroundSource, segproperty) {
    let type = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    const apiType = this.getApiTypeFromEnableVirtualBackground(enabled, backgroundSource, segproperty, type);
    const jsonParams = {
      enabled: enabled,
      backgroundSource: backgroundSource,
      segproperty: segproperty,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          backgroundSource: backgroundSource,
          segproperty: segproperty,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableVirtualBackground(enabled, backgroundSource, segproperty) {
    let type = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : _AgoraMediaBase.MediaSourceType.PrimaryCameraSource;
    return 'RtcEngine_enableVirtualBackground_6dd8ee4';
  }
  setupRemoteVideo(canvas) {
    const apiType = this.getApiTypeFromSetupRemoteVideo(canvas);
    const jsonParams = {
      canvas: canvas,
      toJSON: () => {
        return {
          canvas: canvas
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetupRemoteVideo(canvas) {
    return 'RtcEngine_setupRemoteVideo_acc9c38';
  }
  setupLocalVideo(canvas) {
    const apiType = this.getApiTypeFromSetupLocalVideo(canvas);
    const jsonParams = {
      canvas: canvas,
      toJSON: () => {
        return {
          canvas: canvas
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetupLocalVideo(canvas) {
    return 'RtcEngine_setupLocalVideo_acc9c38';
  }
  setVideoScenario(scenarioType) {
    const apiType = this.getApiTypeFromSetVideoScenario(scenarioType);
    const jsonParams = {
      scenarioType: scenarioType,
      toJSON: () => {
        return {
          scenarioType: scenarioType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVideoScenario(scenarioType) {
    return 'RtcEngine_setVideoScenario_c02cd1c';
  }
  setVideoQoEPreference(qoePreference) {
    const apiType = this.getApiTypeFromSetVideoQoEPreference(qoePreference);
    const jsonParams = {
      qoePreference: qoePreference,
      toJSON: () => {
        return {
          qoePreference: qoePreference
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVideoQoEPreference(qoePreference) {
    return 'RtcEngine_setVideoQoEPreference_c4a3d9f';
  }
  enableAudio() {
    const apiType = this.getApiTypeFromEnableAudio();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAudio() {
    return 'RtcEngine_enableAudio';
  }
  disableAudio() {
    const apiType = this.getApiTypeFromDisableAudio();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDisableAudio() {
    return 'RtcEngine_disableAudio';
  }
  setAudioProfile(profile) {
    let scenario = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraBase.AudioScenarioType.AudioScenarioDefault;
    const apiType = this.getApiTypeFromSetAudioProfile(profile, scenario);
    const jsonParams = {
      profile: profile,
      scenario: scenario,
      toJSON: () => {
        return {
          profile: profile,
          scenario: scenario
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioProfile(profile) {
    let scenario = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraBase.AudioScenarioType.AudioScenarioDefault;
    return 'RtcEngine_setAudioProfile_d944543';
  }
  setAudioScenario(scenario) {
    const apiType = this.getApiTypeFromSetAudioScenario(scenario);
    const jsonParams = {
      scenario: scenario,
      toJSON: () => {
        return {
          scenario: scenario
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioScenario(scenario) {
    return 'RtcEngine_setAudioScenario_c36f5c1';
  }
  enableLocalAudio(enabled) {
    const apiType = this.getApiTypeFromEnableLocalAudio(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableLocalAudio(enabled) {
    return 'RtcEngine_enableLocalAudio_5039d15';
  }
  muteLocalAudioStream(mute) {
    const apiType = this.getApiTypeFromMuteLocalAudioStream(mute);
    const jsonParams = {
      mute: mute,
      toJSON: () => {
        return {
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteLocalAudioStream(mute) {
    return 'RtcEngine_muteLocalAudioStream_5039d15';
  }
  muteAllRemoteAudioStreams(mute) {
    const apiType = this.getApiTypeFromMuteAllRemoteAudioStreams(mute);
    const jsonParams = {
      mute: mute,
      toJSON: () => {
        return {
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteAllRemoteAudioStreams(mute) {
    return 'RtcEngine_muteAllRemoteAudioStreams_5039d15';
  }
  muteRemoteAudioStream(uid, mute) {
    const apiType = this.getApiTypeFromMuteRemoteAudioStream(uid, mute);
    const jsonParams = {
      uid: uid,
      mute: mute,
      toJSON: () => {
        return {
          uid: uid,
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRemoteAudioStream(uid, mute) {
    return 'RtcEngine_muteRemoteAudioStream_dbdc15a';
  }
  muteLocalVideoStream(mute) {
    const apiType = this.getApiTypeFromMuteLocalVideoStream(mute);
    const jsonParams = {
      mute: mute,
      toJSON: () => {
        return {
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteLocalVideoStream(mute) {
    return 'RtcEngine_muteLocalVideoStream_5039d15';
  }
  enableLocalVideo(enabled) {
    const apiType = this.getApiTypeFromEnableLocalVideo(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableLocalVideo(enabled) {
    return 'RtcEngine_enableLocalVideo_5039d15';
  }
  muteAllRemoteVideoStreams(mute) {
    const apiType = this.getApiTypeFromMuteAllRemoteVideoStreams(mute);
    const jsonParams = {
      mute: mute,
      toJSON: () => {
        return {
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteAllRemoteVideoStreams(mute) {
    return 'RtcEngine_muteAllRemoteVideoStreams_5039d15';
  }
  setRemoteDefaultVideoStreamType(streamType) {
    const apiType = this.getApiTypeFromSetRemoteDefaultVideoStreamType(streamType);
    const jsonParams = {
      streamType: streamType,
      toJSON: () => {
        return {
          streamType: streamType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteDefaultVideoStreamType(streamType) {
    return 'RtcEngine_setRemoteDefaultVideoStreamType_5a94eb0';
  }
  muteRemoteVideoStream(uid, mute) {
    const apiType = this.getApiTypeFromMuteRemoteVideoStream(uid, mute);
    const jsonParams = {
      uid: uid,
      mute: mute,
      toJSON: () => {
        return {
          uid: uid,
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRemoteVideoStream(uid, mute) {
    return 'RtcEngine_muteRemoteVideoStream_dbdc15a';
  }
  setRemoteVideoStreamType(uid, streamType) {
    const apiType = this.getApiTypeFromSetRemoteVideoStreamType(uid, streamType);
    const jsonParams = {
      uid: uid,
      streamType: streamType,
      toJSON: () => {
        return {
          uid: uid,
          streamType: streamType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVideoStreamType(uid, streamType) {
    return 'RtcEngine_setRemoteVideoStreamType_9e6406e';
  }
  setRemoteVideoSubscriptionOptions(uid, options) {
    const apiType = this.getApiTypeFromSetRemoteVideoSubscriptionOptions(uid, options);
    const jsonParams = {
      uid: uid,
      options: options,
      toJSON: () => {
        return {
          uid: uid,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVideoSubscriptionOptions(uid, options) {
    return 'RtcEngine_setRemoteVideoSubscriptionOptions_0b6b258';
  }
  setSubscribeAudioBlocklist(uidList, uidNumber) {
    const apiType = this.getApiTypeFromSetSubscribeAudioBlocklist(uidList, uidNumber);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeAudioBlocklist(uidList, uidNumber) {
    return 'RtcEngine_setSubscribeAudioBlocklist_2d31fd5';
  }
  setSubscribeAudioAllowlist(uidList, uidNumber) {
    const apiType = this.getApiTypeFromSetSubscribeAudioAllowlist(uidList, uidNumber);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeAudioAllowlist(uidList, uidNumber) {
    return 'RtcEngine_setSubscribeAudioAllowlist_2d31fd5';
  }
  setSubscribeVideoBlocklist(uidList, uidNumber) {
    const apiType = this.getApiTypeFromSetSubscribeVideoBlocklist(uidList, uidNumber);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeVideoBlocklist(uidList, uidNumber) {
    return 'RtcEngine_setSubscribeVideoBlocklist_2d31fd5';
  }
  setSubscribeVideoAllowlist(uidList, uidNumber) {
    const apiType = this.getApiTypeFromSetSubscribeVideoAllowlist(uidList, uidNumber);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeVideoAllowlist(uidList, uidNumber) {
    return 'RtcEngine_setSubscribeVideoAllowlist_2d31fd5';
  }
  enableAudioVolumeIndication(interval, smooth, reportVad) {
    const apiType = this.getApiTypeFromEnableAudioVolumeIndication(interval, smooth, reportVad);
    const jsonParams = {
      interval: interval,
      smooth: smooth,
      reportVad: reportVad,
      toJSON: () => {
        return {
          interval: interval,
          smooth: smooth,
          reportVad: reportVad
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAudioVolumeIndication(interval, smooth, reportVad) {
    return 'RtcEngine_enableAudioVolumeIndication_39794a0';
  }
  startAudioRecording(config) {
    const apiType = this.getApiTypeFromStartAudioRecording(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartAudioRecording(config) {
    return 'RtcEngine_startAudioRecording_e32bb3b';
  }
  registerAudioEncodedFrameObserver(config, observer) {
    const apiType = this.getApiTypeFromRegisterAudioEncodedFrameObserver(config, observer);
    const jsonParams = {
      config: config,
      observer: observer,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterAudioEncodedFrameObserver(config, observer) {
    return 'RtcEngine_registerAudioEncodedFrameObserver_ed4a177';
  }
  stopAudioRecording() {
    const apiType = this.getApiTypeFromStopAudioRecording();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopAudioRecording() {
    return 'RtcEngine_stopAudioRecording';
  }
  createMediaPlayer() {
    const apiType = this.getApiTypeFromCreateMediaPlayer();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateMediaPlayer() {
    return 'RtcEngine_createMediaPlayer';
  }
  destroyMediaPlayer(mediaPlayer) {
    const apiType = this.getApiTypeFromDestroyMediaPlayer(mediaPlayer);
    const jsonParams = {
      media_player: mediaPlayer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyMediaPlayer(mediaPlayer) {
    return 'RtcEngine_destroyMediaPlayer_328a49b';
  }
  createMediaRecorder(info) {
    const apiType = this.getApiTypeFromCreateMediaRecorder(info);
    const jsonParams = {
      info: info,
      toJSON: () => {
        return {
          info: info
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateMediaRecorder(info) {
    return 'RtcEngine_createMediaRecorder_f779617';
  }
  destroyMediaRecorder(mediaRecorder) {
    const apiType = this.getApiTypeFromDestroyMediaRecorder(mediaRecorder);
    const jsonParams = {
      mediaRecorder: mediaRecorder,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyMediaRecorder(mediaRecorder) {
    return 'RtcEngine_destroyMediaRecorder_95cdef5';
  }
  startAudioMixing(filePath, loopback, cycle) {
    let startPos = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 0;
    const apiType = this.getApiTypeFromStartAudioMixing(filePath, loopback, cycle, startPos);
    const jsonParams = {
      filePath: filePath,
      loopback: loopback,
      cycle: cycle,
      startPos: startPos,
      toJSON: () => {
        return {
          filePath: filePath,
          loopback: loopback,
          cycle: cycle,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartAudioMixing(filePath, loopback, cycle) {
    let startPos = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 0;
    return 'RtcEngine_startAudioMixing_1ee1b1e';
  }
  stopAudioMixing() {
    const apiType = this.getApiTypeFromStopAudioMixing();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopAudioMixing() {
    return 'RtcEngine_stopAudioMixing';
  }
  pauseAudioMixing() {
    const apiType = this.getApiTypeFromPauseAudioMixing();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseAudioMixing() {
    return 'RtcEngine_pauseAudioMixing';
  }
  resumeAudioMixing() {
    const apiType = this.getApiTypeFromResumeAudioMixing();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeAudioMixing() {
    return 'RtcEngine_resumeAudioMixing';
  }
  selectAudioTrack(index) {
    const apiType = this.getApiTypeFromSelectAudioTrack(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSelectAudioTrack(index) {
    return 'RtcEngine_selectAudioTrack_46f8ab7';
  }
  getAudioTrackCount() {
    const apiType = this.getApiTypeFromGetAudioTrackCount();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioTrackCount() {
    return 'RtcEngine_getAudioTrackCount';
  }
  adjustAudioMixingVolume(volume) {
    const apiType = this.getApiTypeFromAdjustAudioMixingVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustAudioMixingVolume(volume) {
    return 'RtcEngine_adjustAudioMixingVolume_46f8ab7';
  }
  adjustAudioMixingPublishVolume(volume) {
    const apiType = this.getApiTypeFromAdjustAudioMixingPublishVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustAudioMixingPublishVolume(volume) {
    return 'RtcEngine_adjustAudioMixingPublishVolume_46f8ab7';
  }
  getAudioMixingPublishVolume() {
    const apiType = this.getApiTypeFromGetAudioMixingPublishVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioMixingPublishVolume() {
    return 'RtcEngine_getAudioMixingPublishVolume';
  }
  adjustAudioMixingPlayoutVolume(volume) {
    const apiType = this.getApiTypeFromAdjustAudioMixingPlayoutVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustAudioMixingPlayoutVolume(volume) {
    return 'RtcEngine_adjustAudioMixingPlayoutVolume_46f8ab7';
  }
  getAudioMixingPlayoutVolume() {
    const apiType = this.getApiTypeFromGetAudioMixingPlayoutVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioMixingPlayoutVolume() {
    return 'RtcEngine_getAudioMixingPlayoutVolume';
  }
  getAudioMixingDuration() {
    const apiType = this.getApiTypeFromGetAudioMixingDuration();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioMixingDuration() {
    return 'RtcEngine_getAudioMixingDuration';
  }
  getAudioMixingCurrentPosition() {
    const apiType = this.getApiTypeFromGetAudioMixingCurrentPosition();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioMixingCurrentPosition() {
    return 'RtcEngine_getAudioMixingCurrentPosition';
  }
  setAudioMixingPosition(pos) {
    const apiType = this.getApiTypeFromSetAudioMixingPosition(pos);
    const jsonParams = {
      pos: pos,
      toJSON: () => {
        return {
          pos: pos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioMixingPosition(pos) {
    return 'RtcEngine_setAudioMixingPosition_46f8ab7';
  }
  setAudioMixingDualMonoMode(mode) {
    const apiType = this.getApiTypeFromSetAudioMixingDualMonoMode(mode);
    const jsonParams = {
      mode: mode,
      toJSON: () => {
        return {
          mode: mode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioMixingDualMonoMode(mode) {
    return 'RtcEngine_setAudioMixingDualMonoMode_38a5515';
  }
  setAudioMixingPitch(pitch) {
    const apiType = this.getApiTypeFromSetAudioMixingPitch(pitch);
    const jsonParams = {
      pitch: pitch,
      toJSON: () => {
        return {
          pitch: pitch
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioMixingPitch(pitch) {
    return 'RtcEngine_setAudioMixingPitch_46f8ab7';
  }
  setAudioMixingPlaybackSpeed(speed) {
    const apiType = this.getApiTypeFromSetAudioMixingPlaybackSpeed(speed);
    const jsonParams = {
      speed: speed,
      toJSON: () => {
        return {
          speed: speed
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioMixingPlaybackSpeed(speed) {
    return 'RtcEngine_setAudioMixingPlaybackSpeed_46f8ab7';
  }
  getEffectsVolume() {
    const apiType = this.getApiTypeFromGetEffectsVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetEffectsVolume() {
    return 'RtcEngine_getEffectsVolume';
  }
  setEffectsVolume(volume) {
    const apiType = this.getApiTypeFromSetEffectsVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetEffectsVolume(volume) {
    return 'RtcEngine_setEffectsVolume_46f8ab7';
  }
  preloadEffect(soundId, filePath) {
    let startPos = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
    const apiType = this.getApiTypeFromPreloadEffect(soundId, filePath, startPos);
    const jsonParams = {
      soundId: soundId,
      filePath: filePath,
      startPos: startPos,
      toJSON: () => {
        return {
          soundId: soundId,
          filePath: filePath,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPreloadEffect(soundId, filePath) {
    let startPos = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
    return 'RtcEngine_preloadEffect_282ba8c';
  }
  playEffect(soundId, filePath, loopCount, pitch, pan, gain) {
    let publish = arguments.length > 6 && arguments[6] !== undefined ? arguments[6] : false;
    let startPos = arguments.length > 7 && arguments[7] !== undefined ? arguments[7] : 0;
    const apiType = this.getApiTypeFromPlayEffect(soundId, filePath, loopCount, pitch, pan, gain, publish, startPos);
    const jsonParams = {
      soundId: soundId,
      filePath: filePath,
      loopCount: loopCount,
      pitch: pitch,
      pan: pan,
      gain: gain,
      publish: publish,
      startPos: startPos,
      toJSON: () => {
        return {
          soundId: soundId,
          filePath: filePath,
          loopCount: loopCount,
          pitch: pitch,
          pan: pan,
          gain: gain,
          publish: publish,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPlayEffect(soundId, filePath, loopCount, pitch, pan, gain) {
    let publish = arguments.length > 6 && arguments[6] !== undefined ? arguments[6] : false;
    let startPos = arguments.length > 7 && arguments[7] !== undefined ? arguments[7] : 0;
    return 'RtcEngine_playEffect_531a783';
  }
  playAllEffects(loopCount, pitch, pan, gain) {
    let publish = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : false;
    const apiType = this.getApiTypeFromPlayAllEffects(loopCount, pitch, pan, gain, publish);
    const jsonParams = {
      loopCount: loopCount,
      pitch: pitch,
      pan: pan,
      gain: gain,
      publish: publish,
      toJSON: () => {
        return {
          loopCount: loopCount,
          pitch: pitch,
          pan: pan,
          gain: gain,
          publish: publish
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPlayAllEffects(loopCount, pitch, pan, gain) {
    let publish = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : false;
    return 'RtcEngine_playAllEffects_20d7df2';
  }
  getVolumeOfEffect(soundId) {
    const apiType = this.getApiTypeFromGetVolumeOfEffect(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetVolumeOfEffect(soundId) {
    return 'RtcEngine_getVolumeOfEffect_46f8ab7';
  }
  setVolumeOfEffect(soundId, volume) {
    const apiType = this.getApiTypeFromSetVolumeOfEffect(soundId, volume);
    const jsonParams = {
      soundId: soundId,
      volume: volume,
      toJSON: () => {
        return {
          soundId: soundId,
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVolumeOfEffect(soundId, volume) {
    return 'RtcEngine_setVolumeOfEffect_4e92b3c';
  }
  pauseEffect(soundId) {
    const apiType = this.getApiTypeFromPauseEffect(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseEffect(soundId) {
    return 'RtcEngine_pauseEffect_46f8ab7';
  }
  pauseAllEffects() {
    const apiType = this.getApiTypeFromPauseAllEffects();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseAllEffects() {
    return 'RtcEngine_pauseAllEffects';
  }
  resumeEffect(soundId) {
    const apiType = this.getApiTypeFromResumeEffect(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeEffect(soundId) {
    return 'RtcEngine_resumeEffect_46f8ab7';
  }
  resumeAllEffects() {
    const apiType = this.getApiTypeFromResumeAllEffects();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeAllEffects() {
    return 'RtcEngine_resumeAllEffects';
  }
  stopEffect(soundId) {
    const apiType = this.getApiTypeFromStopEffect(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopEffect(soundId) {
    return 'RtcEngine_stopEffect_46f8ab7';
  }
  stopAllEffects() {
    const apiType = this.getApiTypeFromStopAllEffects();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopAllEffects() {
    return 'RtcEngine_stopAllEffects';
  }
  unloadEffect(soundId) {
    const apiType = this.getApiTypeFromUnloadEffect(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnloadEffect(soundId) {
    return 'RtcEngine_unloadEffect_46f8ab7';
  }
  unloadAllEffects() {
    const apiType = this.getApiTypeFromUnloadAllEffects();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnloadAllEffects() {
    return 'RtcEngine_unloadAllEffects';
  }
  getEffectDuration(filePath) {
    const apiType = this.getApiTypeFromGetEffectDuration(filePath);
    const jsonParams = {
      filePath: filePath,
      toJSON: () => {
        return {
          filePath: filePath
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetEffectDuration(filePath) {
    return 'RtcEngine_getEffectDuration_3a2037f';
  }
  setEffectPosition(soundId, pos) {
    const apiType = this.getApiTypeFromSetEffectPosition(soundId, pos);
    const jsonParams = {
      soundId: soundId,
      pos: pos,
      toJSON: () => {
        return {
          soundId: soundId,
          pos: pos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetEffectPosition(soundId, pos) {
    return 'RtcEngine_setEffectPosition_4e92b3c';
  }
  getEffectCurrentPosition(soundId) {
    const apiType = this.getApiTypeFromGetEffectCurrentPosition(soundId);
    const jsonParams = {
      soundId: soundId,
      toJSON: () => {
        return {
          soundId: soundId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetEffectCurrentPosition(soundId) {
    return 'RtcEngine_getEffectCurrentPosition_46f8ab7';
  }
  enableSoundPositionIndication(enabled) {
    const apiType = this.getApiTypeFromEnableSoundPositionIndication(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableSoundPositionIndication(enabled) {
    return 'RtcEngine_enableSoundPositionIndication_5039d15';
  }
  setRemoteVoicePosition(uid, pan, gain) {
    const apiType = this.getApiTypeFromSetRemoteVoicePosition(uid, pan, gain);
    const jsonParams = {
      uid: uid,
      pan: pan,
      gain: gain,
      toJSON: () => {
        return {
          uid: uid,
          pan: pan,
          gain: gain
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVoicePosition(uid, pan, gain) {
    return 'RtcEngine_setRemoteVoicePosition_250b42d';
  }
  enableSpatialAudio(enabled) {
    const apiType = this.getApiTypeFromEnableSpatialAudio(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableSpatialAudio(enabled) {
    return 'RtcEngine_enableSpatialAudio_5039d15';
  }
  setRemoteUserSpatialAudioParams(uid, params) {
    const apiType = this.getApiTypeFromSetRemoteUserSpatialAudioParams(uid, params);
    const jsonParams = {
      uid: uid,
      params: params,
      toJSON: () => {
        return {
          uid: uid,
          params: params
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteUserSpatialAudioParams(uid, params) {
    return 'RtcEngine_setRemoteUserSpatialAudioParams_65a7855';
  }
  setVoiceBeautifierPreset(preset) {
    const apiType = this.getApiTypeFromSetVoiceBeautifierPreset(preset);
    const jsonParams = {
      preset: preset,
      toJSON: () => {
        return {
          preset: preset
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVoiceBeautifierPreset(preset) {
    return 'RtcEngine_setVoiceBeautifierPreset_4dd6319';
  }
  setAudioEffectPreset(preset) {
    const apiType = this.getApiTypeFromSetAudioEffectPreset(preset);
    const jsonParams = {
      preset: preset,
      toJSON: () => {
        return {
          preset: preset
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioEffectPreset(preset) {
    return 'RtcEngine_setAudioEffectPreset_92ea92c';
  }
  setVoiceConversionPreset(preset) {
    const apiType = this.getApiTypeFromSetVoiceConversionPreset(preset);
    const jsonParams = {
      preset: preset,
      toJSON: () => {
        return {
          preset: preset
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVoiceConversionPreset(preset) {
    return 'RtcEngine_setVoiceConversionPreset_d14ee73';
  }
  setAudioEffectParameters(preset, param1, param2) {
    const apiType = this.getApiTypeFromSetAudioEffectParameters(preset, param1, param2);
    const jsonParams = {
      preset: preset,
      param1: param1,
      param2: param2,
      toJSON: () => {
        return {
          preset: preset,
          param1: param1,
          param2: param2
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioEffectParameters(preset, param1, param2) {
    return 'RtcEngine_setAudioEffectParameters_73bc670';
  }
  setVoiceBeautifierParameters(preset, param1, param2) {
    const apiType = this.getApiTypeFromSetVoiceBeautifierParameters(preset, param1, param2);
    const jsonParams = {
      preset: preset,
      param1: param1,
      param2: param2,
      toJSON: () => {
        return {
          preset: preset,
          param1: param1,
          param2: param2
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVoiceBeautifierParameters(preset, param1, param2) {
    return 'RtcEngine_setVoiceBeautifierParameters_f3cf745';
  }
  setVoiceConversionParameters(preset, param1, param2) {
    const apiType = this.getApiTypeFromSetVoiceConversionParameters(preset, param1, param2);
    const jsonParams = {
      preset: preset,
      param1: param1,
      param2: param2,
      toJSON: () => {
        return {
          preset: preset,
          param1: param1,
          param2: param2
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVoiceConversionParameters(preset, param1, param2) {
    return 'RtcEngine_setVoiceConversionParameters_2f5022e';
  }
  setLocalVoicePitch(pitch) {
    const apiType = this.getApiTypeFromSetLocalVoicePitch(pitch);
    const jsonParams = {
      pitch: pitch,
      toJSON: () => {
        return {
          pitch: pitch
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalVoicePitch(pitch) {
    return 'RtcEngine_setLocalVoicePitch_bdb36bb';
  }
  setLocalVoiceFormant(formantRatio) {
    const apiType = this.getApiTypeFromSetLocalVoiceFormant(formantRatio);
    const jsonParams = {
      formantRatio: formantRatio,
      toJSON: () => {
        return {
          formantRatio: formantRatio
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalVoiceFormant(formantRatio) {
    return 'RtcEngine_setLocalVoiceFormant_bdb36bb';
  }
  setLocalVoiceEqualization(bandFrequency, bandGain) {
    const apiType = this.getApiTypeFromSetLocalVoiceEqualization(bandFrequency, bandGain);
    const jsonParams = {
      bandFrequency: bandFrequency,
      bandGain: bandGain,
      toJSON: () => {
        return {
          bandFrequency: bandFrequency,
          bandGain: bandGain
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalVoiceEqualization(bandFrequency, bandGain) {
    return 'RtcEngine_setLocalVoiceEqualization_d14012c';
  }
  setLocalVoiceReverb(reverbKey, value) {
    const apiType = this.getApiTypeFromSetLocalVoiceReverb(reverbKey, value);
    const jsonParams = {
      reverbKey: reverbKey,
      value: value,
      toJSON: () => {
        return {
          reverbKey: reverbKey,
          value: value
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalVoiceReverb(reverbKey, value) {
    return 'RtcEngine_setLocalVoiceReverb_29c2013';
  }
  setHeadphoneEQPreset(preset) {
    const apiType = this.getApiTypeFromSetHeadphoneEQPreset(preset);
    const jsonParams = {
      preset: preset,
      toJSON: () => {
        return {
          preset: preset
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetHeadphoneEQPreset(preset) {
    return 'RtcEngine_setHeadphoneEQPreset_b679644';
  }
  setHeadphoneEQParameters(lowGain, highGain) {
    const apiType = this.getApiTypeFromSetHeadphoneEQParameters(lowGain, highGain);
    const jsonParams = {
      lowGain: lowGain,
      highGain: highGain,
      toJSON: () => {
        return {
          lowGain: lowGain,
          highGain: highGain
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetHeadphoneEQParameters(lowGain, highGain) {
    return 'RtcEngine_setHeadphoneEQParameters_4e92b3c';
  }
  enableVoiceAITuner(enabled, type) {
    const apiType = this.getApiTypeFromEnableVoiceAITuner(enabled, type);
    const jsonParams = {
      enabled: enabled,
      type: type,
      toJSON: () => {
        return {
          enabled: enabled,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableVoiceAITuner(enabled, type) {
    return 'RtcEngine_enableVoiceAITuner_28f5d5b';
  }
  setLogFile(filePath) {
    const apiType = this.getApiTypeFromSetLogFile(filePath);
    const jsonParams = {
      filePath: filePath,
      toJSON: () => {
        return {
          filePath: filePath
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLogFile(filePath) {
    return 'RtcEngine_setLogFile_3a2037f';
  }
  setLogFilter(filter) {
    const apiType = this.getApiTypeFromSetLogFilter(filter);
    const jsonParams = {
      filter: filter,
      toJSON: () => {
        return {
          filter: filter
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLogFilter(filter) {
    return 'RtcEngine_setLogFilter_2626ac7';
  }
  setLogLevel(level) {
    const apiType = this.getApiTypeFromSetLogLevel(level);
    const jsonParams = {
      level: level,
      toJSON: () => {
        return {
          level: level
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLogLevel(level) {
    return 'RtcEngine_setLogLevel_f125d83';
  }
  setLogFileSize(fileSizeInKBytes) {
    const apiType = this.getApiTypeFromSetLogFileSize(fileSizeInKBytes);
    const jsonParams = {
      fileSizeInKBytes: fileSizeInKBytes,
      toJSON: () => {
        return {
          fileSizeInKBytes: fileSizeInKBytes
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLogFileSize(fileSizeInKBytes) {
    return 'RtcEngine_setLogFileSize_2626ac7';
  }
  uploadLogFile() {
    const apiType = this.getApiTypeFromUploadLogFile();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const requestId = jsonResults.requestId;
    return requestId;
  }
  getApiTypeFromUploadLogFile() {
    return 'RtcEngine_uploadLogFile_66d4ecd';
  }
  writeLog(level, fmt) {
    const apiType = this.getApiTypeFromWriteLog(level, fmt);
    const jsonParams = {
      level: level,
      fmt: fmt,
      toJSON: () => {
        return {
          level: level,
          fmt: fmt
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromWriteLog(level, fmt) {
    return 'RtcEngine_writeLog_62889f6';
  }
  setLocalRenderMode(renderMode) {
    let mirrorMode = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraBase.VideoMirrorModeType.VideoMirrorModeAuto;
    const apiType = this.getApiTypeFromSetLocalRenderMode(renderMode, mirrorMode);
    const jsonParams = {
      renderMode: renderMode,
      mirrorMode: mirrorMode,
      toJSON: () => {
        return {
          renderMode: renderMode,
          mirrorMode: mirrorMode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalRenderMode(renderMode) {
    let mirrorMode = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraBase.VideoMirrorModeType.VideoMirrorModeAuto;
    return 'RtcEngine_setLocalRenderMode_cfb201b';
  }
  setRemoteRenderMode(uid, renderMode, mirrorMode) {
    const apiType = this.getApiTypeFromSetRemoteRenderMode(uid, renderMode, mirrorMode);
    const jsonParams = {
      uid: uid,
      renderMode: renderMode,
      mirrorMode: mirrorMode,
      toJSON: () => {
        return {
          uid: uid,
          renderMode: renderMode,
          mirrorMode: mirrorMode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteRenderMode(uid, renderMode, mirrorMode) {
    return 'RtcEngine_setRemoteRenderMode_6771ce0';
  }
  setLocalRenderTargetFps(sourceType, targetFps) {
    const apiType = this.getApiTypeFromSetLocalRenderTargetFps(sourceType, targetFps);
    const jsonParams = {
      sourceType: sourceType,
      targetFps: targetFps,
      toJSON: () => {
        return {
          sourceType: sourceType,
          targetFps: targetFps
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalRenderTargetFps(sourceType, targetFps) {
    return 'RtcEngine_setLocalRenderTargetFps_2ad83d8';
  }
  setRemoteRenderTargetFps(targetFps) {
    const apiType = this.getApiTypeFromSetRemoteRenderTargetFps(targetFps);
    const jsonParams = {
      targetFps: targetFps,
      toJSON: () => {
        return {
          targetFps: targetFps
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteRenderTargetFps(targetFps) {
    return 'RtcEngine_setRemoteRenderTargetFps_46f8ab7';
  }
  setLocalVideoMirrorMode(mirrorMode) {
    const apiType = this.getApiTypeFromSetLocalVideoMirrorMode(mirrorMode);
    const jsonParams = {
      mirrorMode: mirrorMode,
      toJSON: () => {
        return {
          mirrorMode: mirrorMode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalVideoMirrorMode(mirrorMode) {
    return 'RtcEngine_setLocalVideoMirrorMode_b8a6c69';
  }
  enableDualStreamMode(enabled, streamConfig) {
    const apiType = this.getApiTypeFromEnableDualStreamMode(enabled, streamConfig);
    const jsonParams = {
      enabled: enabled,
      streamConfig: streamConfig,
      toJSON: () => {
        return {
          enabled: enabled,
          streamConfig: streamConfig
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableDualStreamMode(enabled, streamConfig) {
    return 'RtcEngine_enableDualStreamMode_9822d8a';
  }
  setDualStreamMode(mode, streamConfig) {
    const apiType = this.getApiTypeFromSetDualStreamMode(mode, streamConfig);
    const jsonParams = {
      mode: mode,
      streamConfig: streamConfig,
      toJSON: () => {
        return {
          mode: mode,
          streamConfig: streamConfig
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDualStreamMode(mode, streamConfig) {
    return 'RtcEngine_setDualStreamMode_b3a4f6c';
  }
  setSimulcastConfig(simulcastConfig) {
    const apiType = this.getApiTypeFromSetSimulcastConfig(simulcastConfig);
    const jsonParams = {
      simulcastConfig: simulcastConfig,
      toJSON: () => {
        return {
          simulcastConfig: simulcastConfig
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSimulcastConfig(simulcastConfig) {
    return 'RtcEngine_setSimulcastConfig_3dcdfd7';
  }
  enableCustomAudioLocalPlayback(trackId, enabled) {
    const apiType = this.getApiTypeFromEnableCustomAudioLocalPlayback(trackId, enabled);
    const jsonParams = {
      trackId: trackId,
      enabled: enabled,
      toJSON: () => {
        return {
          trackId: trackId,
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableCustomAudioLocalPlayback(trackId, enabled) {
    return 'RtcEngine_enableCustomAudioLocalPlayback_9566341';
  }
  setRecordingAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    const apiType = this.getApiTypeFromSetRecordingAudioFrameParameters(sampleRate, channel, mode, samplesPerCall);
    const jsonParams = {
      sampleRate: sampleRate,
      channel: channel,
      mode: mode,
      samplesPerCall: samplesPerCall,
      toJSON: () => {
        return {
          sampleRate: sampleRate,
          channel: channel,
          mode: mode,
          samplesPerCall: samplesPerCall
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRecordingAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    return 'RtcEngine_setRecordingAudioFrameParameters_bd46d1d';
  }
  setPlaybackAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    const apiType = this.getApiTypeFromSetPlaybackAudioFrameParameters(sampleRate, channel, mode, samplesPerCall);
    const jsonParams = {
      sampleRate: sampleRate,
      channel: channel,
      mode: mode,
      samplesPerCall: samplesPerCall,
      toJSON: () => {
        return {
          sampleRate: sampleRate,
          channel: channel,
          mode: mode,
          samplesPerCall: samplesPerCall
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlaybackAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    return 'RtcEngine_setPlaybackAudioFrameParameters_bd46d1d';
  }
  setMixedAudioFrameParameters(sampleRate, channel, samplesPerCall) {
    const apiType = this.getApiTypeFromSetMixedAudioFrameParameters(sampleRate, channel, samplesPerCall);
    const jsonParams = {
      sampleRate: sampleRate,
      channel: channel,
      samplesPerCall: samplesPerCall,
      toJSON: () => {
        return {
          sampleRate: sampleRate,
          channel: channel,
          samplesPerCall: samplesPerCall
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetMixedAudioFrameParameters(sampleRate, channel, samplesPerCall) {
    return 'RtcEngine_setMixedAudioFrameParameters_ee7e270';
  }
  setEarMonitoringAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    const apiType = this.getApiTypeFromSetEarMonitoringAudioFrameParameters(sampleRate, channel, mode, samplesPerCall);
    const jsonParams = {
      sampleRate: sampleRate,
      channel: channel,
      mode: mode,
      samplesPerCall: samplesPerCall,
      toJSON: () => {
        return {
          sampleRate: sampleRate,
          channel: channel,
          mode: mode,
          samplesPerCall: samplesPerCall
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetEarMonitoringAudioFrameParameters(sampleRate, channel, mode, samplesPerCall) {
    return 'RtcEngine_setEarMonitoringAudioFrameParameters_bd46d1d';
  }
  setPlaybackAudioFrameBeforeMixingParameters(sampleRate, channel) {
    const apiType = this.getApiTypeFromSetPlaybackAudioFrameBeforeMixingParameters(sampleRate, channel);
    const jsonParams = {
      sampleRate: sampleRate,
      channel: channel,
      toJSON: () => {
        return {
          sampleRate: sampleRate,
          channel: channel
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlaybackAudioFrameBeforeMixingParameters(sampleRate, channel) {
    return 'RtcEngine_setPlaybackAudioFrameBeforeMixingParameters_4e92b3c';
  }
  enableAudioSpectrumMonitor() {
    let intervalInMS = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 100;
    const apiType = this.getApiTypeFromEnableAudioSpectrumMonitor(intervalInMS);
    const jsonParams = {
      intervalInMS: intervalInMS,
      toJSON: () => {
        return {
          intervalInMS: intervalInMS
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAudioSpectrumMonitor() {
    let intervalInMS = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 100;
    return 'RtcEngine_enableAudioSpectrumMonitor_46f8ab7';
  }
  disableAudioSpectrumMonitor() {
    const apiType = this.getApiTypeFromDisableAudioSpectrumMonitor();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDisableAudioSpectrumMonitor() {
    return 'RtcEngine_disableAudioSpectrumMonitor';
  }
  registerAudioSpectrumObserver(observer) {
    const apiType = this.getApiTypeFromRegisterAudioSpectrumObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterAudioSpectrumObserver(observer) {
    return 'RtcEngine_registerAudioSpectrumObserver_0406ea7';
  }
  unregisterAudioSpectrumObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterAudioSpectrumObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterAudioSpectrumObserver(observer) {
    return 'RtcEngine_unregisterAudioSpectrumObserver_0406ea7';
  }
  adjustRecordingSignalVolume(volume) {
    const apiType = this.getApiTypeFromAdjustRecordingSignalVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustRecordingSignalVolume(volume) {
    return 'RtcEngine_adjustRecordingSignalVolume_46f8ab7';
  }
  muteRecordingSignal(mute) {
    const apiType = this.getApiTypeFromMuteRecordingSignal(mute);
    const jsonParams = {
      mute: mute,
      toJSON: () => {
        return {
          mute: mute
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRecordingSignal(mute) {
    return 'RtcEngine_muteRecordingSignal_5039d15';
  }
  adjustPlaybackSignalVolume(volume) {
    const apiType = this.getApiTypeFromAdjustPlaybackSignalVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustPlaybackSignalVolume(volume) {
    return 'RtcEngine_adjustPlaybackSignalVolume_46f8ab7';
  }
  adjustUserPlaybackSignalVolume(uid, volume) {
    const apiType = this.getApiTypeFromAdjustUserPlaybackSignalVolume(uid, volume);
    const jsonParams = {
      uid: uid,
      volume: volume,
      toJSON: () => {
        return {
          uid: uid,
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustUserPlaybackSignalVolume(uid, volume) {
    return 'RtcEngine_adjustUserPlaybackSignalVolume_88641bf';
  }
  setLocalPublishFallbackOption(option) {
    const apiType = this.getApiTypeFromSetLocalPublishFallbackOption(option);
    const jsonParams = {
      option: option,
      toJSON: () => {
        return {
          option: option
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalPublishFallbackOption(option) {
    return 'RtcEngine_setLocalPublishFallbackOption_c29b788';
  }
  setRemoteSubscribeFallbackOption(option) {
    const apiType = this.getApiTypeFromSetRemoteSubscribeFallbackOption(option);
    const jsonParams = {
      option: option,
      toJSON: () => {
        return {
          option: option
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteSubscribeFallbackOption(option) {
    return 'RtcEngine_setRemoteSubscribeFallbackOption_c29b788';
  }
  setHighPriorityUserList(uidList, uidNum, option) {
    const apiType = this.getApiTypeFromSetHighPriorityUserList(uidList, uidNum, option);
    const jsonParams = {
      uidList: uidList,
      uidNum: uidNum,
      option: option,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNum: uidNum,
          option: option
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetHighPriorityUserList(uidList, uidNum, option) {
    return 'RtcEngine_setHighPriorityUserList_ab88726';
  }
  enableExtension(provider, extension) {
    let enable = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
    let type = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    const apiType = this.getApiTypeFromEnableExtension(provider, extension, enable, type);
    const jsonParams = {
      provider: provider,
      extension: extension,
      enable: enable,
      type: type,
      toJSON: () => {
        return {
          provider: provider,
          extension: extension,
          enable: enable,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableExtension(provider, extension) {
    let enable = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
    let type = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    return 'RtcEngine_enableExtension_0b60a2c';
  }
  setExtensionProperty(provider, extension, key, value) {
    let type = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    const apiType = this.getApiTypeFromSetExtensionProperty(provider, extension, key, value, type);
    const jsonParams = {
      provider: provider,
      extension: extension,
      key: key,
      value: value,
      type: type,
      toJSON: () => {
        return {
          provider: provider,
          extension: extension,
          key: key,
          value: value,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExtensionProperty(provider, extension, key, value) {
    let type = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    return 'RtcEngine_setExtensionProperty_520ac55';
  }
  getExtensionProperty(provider, extension, key, bufLen) {
    let type = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    const apiType = this.getApiTypeFromGetExtensionProperty(provider, extension, key, bufLen, type);
    const jsonParams = {
      provider: provider,
      extension: extension,
      key: key,
      buf_len: bufLen,
      type: type,
      toJSON: () => {
        return {
          provider: provider,
          extension: extension,
          key: key,
          buf_len: bufLen,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const value = jsonResults.value;
    return value;
  }
  getApiTypeFromGetExtensionProperty(provider, extension, key, bufLen) {
    let type = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    return 'RtcEngine_getExtensionProperty_38c9723';
  }
  enableLoopbackRecording(enabled, deviceName) {
    const apiType = this.getApiTypeFromEnableLoopbackRecording(enabled, deviceName);
    const jsonParams = {
      enabled: enabled,
      deviceName: deviceName,
      toJSON: () => {
        return {
          enabled: enabled,
          deviceName: deviceName
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableLoopbackRecording(enabled, deviceName) {
    return 'RtcEngine_enableLoopbackRecording_0b8eb79';
  }
  adjustLoopbackSignalVolume(volume) {
    const apiType = this.getApiTypeFromAdjustLoopbackSignalVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustLoopbackSignalVolume(volume) {
    return 'RtcEngine_adjustLoopbackSignalVolume_46f8ab7';
  }
  getLoopbackRecordingVolume() {
    const apiType = this.getApiTypeFromGetLoopbackRecordingVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetLoopbackRecordingVolume() {
    return 'RtcEngine_getLoopbackRecordingVolume';
  }
  enableInEarMonitoring(enabled, includeAudioFilters) {
    const apiType = this.getApiTypeFromEnableInEarMonitoring(enabled, includeAudioFilters);
    const jsonParams = {
      enabled: enabled,
      includeAudioFilters: includeAudioFilters,
      toJSON: () => {
        return {
          enabled: enabled,
          includeAudioFilters: includeAudioFilters
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableInEarMonitoring(enabled, includeAudioFilters) {
    return 'RtcEngine_enableInEarMonitoring_077cf5f';
  }
  setInEarMonitoringVolume(volume) {
    const apiType = this.getApiTypeFromSetInEarMonitoringVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetInEarMonitoringVolume(volume) {
    return 'RtcEngine_setInEarMonitoringVolume_46f8ab7';
  }
  loadExtensionProvider(path) {
    let unloadAfterUse = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    const apiType = this.getApiTypeFromLoadExtensionProvider(path, unloadAfterUse);
    const jsonParams = {
      path: path,
      unload_after_use: unloadAfterUse,
      toJSON: () => {
        return {
          path: path,
          unload_after_use: unloadAfterUse
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromLoadExtensionProvider(path) {
    let unloadAfterUse = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    return 'RtcEngine_loadExtensionProvider_7a174df';
  }
  setExtensionProviderProperty(provider, key, value) {
    const apiType = this.getApiTypeFromSetExtensionProviderProperty(provider, key, value);
    const jsonParams = {
      provider: provider,
      key: key,
      value: value,
      toJSON: () => {
        return {
          provider: provider,
          key: key,
          value: value
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExtensionProviderProperty(provider, key, value) {
    return 'RtcEngine_setExtensionProviderProperty_0e4f59e';
  }
  registerExtension(provider, extension) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    const apiType = this.getApiTypeFromRegisterExtension(provider, extension, type);
    const jsonParams = {
      provider: provider,
      extension: extension,
      type: type,
      toJSON: () => {
        return {
          provider: provider,
          extension: extension,
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterExtension(provider, extension) {
    let type = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.MediaSourceType.UnknownMediaSource;
    return 'RtcEngine_registerExtension_fd62af4';
  }
  setCameraCapturerConfiguration(config) {
    const apiType = this.getApiTypeFromSetCameraCapturerConfiguration(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraCapturerConfiguration(config) {
    return 'RtcEngine_setCameraCapturerConfiguration_afa93b3';
  }
  createCustomVideoTrack() {
    const apiType = this.getApiTypeFromCreateCustomVideoTrack();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateCustomVideoTrack() {
    return 'RtcEngine_createCustomVideoTrack';
  }
  createCustomEncodedVideoTrack(senderOption) {
    const apiType = this.getApiTypeFromCreateCustomEncodedVideoTrack(senderOption);
    const jsonParams = {
      sender_option: senderOption,
      toJSON: () => {
        return {
          sender_option: senderOption
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateCustomEncodedVideoTrack(senderOption) {
    return 'RtcEngine_createCustomEncodedVideoTrack_0e9dc99';
  }
  destroyCustomVideoTrack(videoTrackId) {
    const apiType = this.getApiTypeFromDestroyCustomVideoTrack(videoTrackId);
    const jsonParams = {
      video_track_id: videoTrackId,
      toJSON: () => {
        return {
          video_track_id: videoTrackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyCustomVideoTrack(videoTrackId) {
    return 'RtcEngine_destroyCustomVideoTrack_3019423';
  }
  destroyCustomEncodedVideoTrack(videoTrackId) {
    const apiType = this.getApiTypeFromDestroyCustomEncodedVideoTrack(videoTrackId);
    const jsonParams = {
      video_track_id: videoTrackId,
      toJSON: () => {
        return {
          video_track_id: videoTrackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyCustomEncodedVideoTrack(videoTrackId) {
    return 'RtcEngine_destroyCustomEncodedVideoTrack_3019423';
  }
  switchCamera() {
    const apiType = this.getApiTypeFromSwitchCamera();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSwitchCamera() {
    return 'RtcEngine_switchCamera';
  }
  isCameraZoomSupported() {
    const apiType = this.getApiTypeFromIsCameraZoomSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraZoomSupported() {
    return 'RtcEngine_isCameraZoomSupported';
  }
  isCameraFaceDetectSupported() {
    const apiType = this.getApiTypeFromIsCameraFaceDetectSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraFaceDetectSupported() {
    return 'RtcEngine_isCameraFaceDetectSupported';
  }
  isCameraTorchSupported() {
    const apiType = this.getApiTypeFromIsCameraTorchSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraTorchSupported() {
    return 'RtcEngine_isCameraTorchSupported';
  }
  isCameraFocusSupported() {
    const apiType = this.getApiTypeFromIsCameraFocusSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraFocusSupported() {
    return 'RtcEngine_isCameraFocusSupported';
  }
  isCameraAutoFocusFaceModeSupported() {
    const apiType = this.getApiTypeFromIsCameraAutoFocusFaceModeSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraAutoFocusFaceModeSupported() {
    return 'RtcEngine_isCameraAutoFocusFaceModeSupported';
  }
  setCameraZoomFactor(factor) {
    const apiType = this.getApiTypeFromSetCameraZoomFactor(factor);
    const jsonParams = {
      factor: factor,
      toJSON: () => {
        return {
          factor: factor
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraZoomFactor(factor) {
    return 'RtcEngine_setCameraZoomFactor_685e803';
  }
  enableFaceDetection(enabled) {
    const apiType = this.getApiTypeFromEnableFaceDetection(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableFaceDetection(enabled) {
    return 'RtcEngine_enableFaceDetection_5039d15';
  }
  getCameraMaxZoomFactor() {
    const apiType = this.getApiTypeFromGetCameraMaxZoomFactor();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCameraMaxZoomFactor() {
    return 'RtcEngine_getCameraMaxZoomFactor';
  }
  setCameraFocusPositionInPreview(positionX, positionY) {
    const apiType = this.getApiTypeFromSetCameraFocusPositionInPreview(positionX, positionY);
    const jsonParams = {
      positionX: positionX,
      positionY: positionY,
      toJSON: () => {
        return {
          positionX: positionX,
          positionY: positionY
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraFocusPositionInPreview(positionX, positionY) {
    return 'RtcEngine_setCameraFocusPositionInPreview_f282d50';
  }
  setCameraTorchOn(isOn) {
    const apiType = this.getApiTypeFromSetCameraTorchOn(isOn);
    const jsonParams = {
      isOn: isOn,
      toJSON: () => {
        return {
          isOn: isOn
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraTorchOn(isOn) {
    return 'RtcEngine_setCameraTorchOn_5039d15';
  }
  setCameraAutoFocusFaceModeEnabled(enabled) {
    const apiType = this.getApiTypeFromSetCameraAutoFocusFaceModeEnabled(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraAutoFocusFaceModeEnabled(enabled) {
    return 'RtcEngine_setCameraAutoFocusFaceModeEnabled_5039d15';
  }
  isCameraExposurePositionSupported() {
    const apiType = this.getApiTypeFromIsCameraExposurePositionSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraExposurePositionSupported() {
    return 'RtcEngine_isCameraExposurePositionSupported';
  }
  setCameraExposurePosition(positionXinView, positionYinView) {
    const apiType = this.getApiTypeFromSetCameraExposurePosition(positionXinView, positionYinView);
    const jsonParams = {
      positionXinView: positionXinView,
      positionYinView: positionYinView,
      toJSON: () => {
        return {
          positionXinView: positionXinView,
          positionYinView: positionYinView
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraExposurePosition(positionXinView, positionYinView) {
    return 'RtcEngine_setCameraExposurePosition_f282d50';
  }
  isCameraExposureSupported() {
    const apiType = this.getApiTypeFromIsCameraExposureSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraExposureSupported() {
    return 'RtcEngine_isCameraExposureSupported';
  }
  setCameraExposureFactor(factor) {
    const apiType = this.getApiTypeFromSetCameraExposureFactor(factor);
    const jsonParams = {
      factor: factor,
      toJSON: () => {
        return {
          factor: factor
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraExposureFactor(factor) {
    return 'RtcEngine_setCameraExposureFactor_685e803';
  }
  isCameraAutoExposureFaceModeSupported() {
    const apiType = this.getApiTypeFromIsCameraAutoExposureFaceModeSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraAutoExposureFaceModeSupported() {
    return 'RtcEngine_isCameraAutoExposureFaceModeSupported';
  }
  setCameraAutoExposureFaceModeEnabled(enabled) {
    const apiType = this.getApiTypeFromSetCameraAutoExposureFaceModeEnabled(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraAutoExposureFaceModeEnabled(enabled) {
    return 'RtcEngine_setCameraAutoExposureFaceModeEnabled_5039d15';
  }
  setCameraStabilizationMode(mode) {
    const apiType = this.getApiTypeFromSetCameraStabilizationMode(mode);
    const jsonParams = {
      mode: mode,
      toJSON: () => {
        return {
          mode: mode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraStabilizationMode(mode) {
    return 'RtcEngine_setCameraStabilizationMode_701b981';
  }
  setDefaultAudioRouteToSpeakerphone(defaultToSpeaker) {
    const apiType = this.getApiTypeFromSetDefaultAudioRouteToSpeakerphone(defaultToSpeaker);
    const jsonParams = {
      defaultToSpeaker: defaultToSpeaker,
      toJSON: () => {
        return {
          defaultToSpeaker: defaultToSpeaker
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDefaultAudioRouteToSpeakerphone(defaultToSpeaker) {
    return 'RtcEngine_setDefaultAudioRouteToSpeakerphone_5039d15';
  }
  setEnableSpeakerphone(speakerOn) {
    const apiType = this.getApiTypeFromSetEnableSpeakerphone(speakerOn);
    const jsonParams = {
      speakerOn: speakerOn,
      toJSON: () => {
        return {
          speakerOn: speakerOn
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetEnableSpeakerphone(speakerOn) {
    return 'RtcEngine_setEnableSpeakerphone_5039d15';
  }
  isSpeakerphoneEnabled() {
    const apiType = this.getApiTypeFromIsSpeakerphoneEnabled();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsSpeakerphoneEnabled() {
    return 'RtcEngine_isSpeakerphoneEnabled';
  }
  setRouteInCommunicationMode(route) {
    const apiType = this.getApiTypeFromSetRouteInCommunicationMode(route);
    const jsonParams = {
      route: route,
      toJSON: () => {
        return {
          route: route
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRouteInCommunicationMode(route) {
    return 'RtcEngine_setRouteInCommunicationMode_46f8ab7';
  }
  isCameraCenterStageSupported() {
    const apiType = this.getApiTypeFromIsCameraCenterStageSupported();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsCameraCenterStageSupported() {
    return 'RtcEngine_isCameraCenterStageSupported';
  }
  enableCameraCenterStage(enabled) {
    const apiType = this.getApiTypeFromEnableCameraCenterStage(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableCameraCenterStage(enabled) {
    return 'RtcEngine_enableCameraCenterStage_5039d15';
  }
  getScreenCaptureSources(thumbSize, iconSize, includeScreen) {
    const apiType = this.getApiTypeFromGetScreenCaptureSources(thumbSize, iconSize, includeScreen);
    const jsonParams = {
      thumbSize: thumbSize,
      iconSize: iconSize,
      includeScreen: includeScreen,
      toJSON: () => {
        return {
          thumbSize: thumbSize,
          iconSize: iconSize,
          includeScreen: includeScreen
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetScreenCaptureSources(thumbSize, iconSize, includeScreen) {
    return 'RtcEngine_getScreenCaptureSources_f3e02cb';
  }
  setAudioSessionOperationRestriction(restriction) {
    const apiType = this.getApiTypeFromSetAudioSessionOperationRestriction(restriction);
    const jsonParams = {
      restriction: restriction,
      toJSON: () => {
        return {
          restriction: restriction
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioSessionOperationRestriction(restriction) {
    return 'RtcEngine_setAudioSessionOperationRestriction_c492897';
  }
  startScreenCaptureByDisplayId(displayId, regionRect, captureParams) {
    const apiType = this.getApiTypeFromStartScreenCaptureByDisplayId(displayId, regionRect, captureParams);
    const jsonParams = {
      displayId: displayId,
      regionRect: regionRect,
      captureParams: captureParams,
      toJSON: () => {
        return {
          displayId: displayId,
          regionRect: regionRect,
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartScreenCaptureByDisplayId(displayId, regionRect, captureParams) {
    return 'RtcEngine_startScreenCaptureByDisplayId_ce89867';
  }
  startScreenCaptureByScreenRect(screenRect, regionRect, captureParams) {
    const apiType = this.getApiTypeFromStartScreenCaptureByScreenRect(screenRect, regionRect, captureParams);
    const jsonParams = {
      screenRect: screenRect,
      regionRect: regionRect,
      captureParams: captureParams,
      toJSON: () => {
        return {
          screenRect: screenRect,
          regionRect: regionRect,
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartScreenCaptureByScreenRect(screenRect, regionRect, captureParams) {
    return 'RtcEngine_startScreenCaptureByScreenRect_e286286';
  }
  getAudioDeviceInfo() {
    const apiType = this.getApiTypeFromGetAudioDeviceInfo();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const deviceInfo = jsonResults.deviceInfo;
    return deviceInfo;
  }
  getApiTypeFromGetAudioDeviceInfo() {
    return 'RtcEngine_getAudioDeviceInfo_505aa0c';
  }
  startScreenCaptureByWindowId(windowId, regionRect, captureParams) {
    const apiType = this.getApiTypeFromStartScreenCaptureByWindowId(windowId, regionRect, captureParams);
    const jsonParams = {
      windowId: windowId,
      regionRect: regionRect,
      captureParams: captureParams,
      toJSON: () => {
        return {
          windowId: windowId,
          regionRect: regionRect,
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartScreenCaptureByWindowId(windowId, regionRect, captureParams) {
    return 'RtcEngine_startScreenCaptureByWindowId_ce89867';
  }
  setScreenCaptureContentHint(contentHint) {
    const apiType = this.getApiTypeFromSetScreenCaptureContentHint(contentHint);
    const jsonParams = {
      contentHint: contentHint,
      toJSON: () => {
        return {
          contentHint: contentHint
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetScreenCaptureContentHint(contentHint) {
    return 'RtcEngine_setScreenCaptureContentHint_8ad2c79';
  }
  updateScreenCaptureRegion(regionRect) {
    const apiType = this.getApiTypeFromUpdateScreenCaptureRegion(regionRect);
    const jsonParams = {
      regionRect: regionRect,
      toJSON: () => {
        return {
          regionRect: regionRect
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateScreenCaptureRegion(regionRect) {
    return 'RtcEngine_updateScreenCaptureRegion_6b327a8';
  }
  updateScreenCaptureParameters(captureParams) {
    const apiType = this.getApiTypeFromUpdateScreenCaptureParameters(captureParams);
    const jsonParams = {
      captureParams: captureParams,
      toJSON: () => {
        return {
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateScreenCaptureParameters(captureParams) {
    return 'RtcEngine_updateScreenCaptureParameters_a2eef93';
  }
  startScreenCapture(captureParams) {
    const apiType = this.getApiTypeFromStartScreenCapture(captureParams);
    const jsonParams = {
      captureParams: captureParams,
      toJSON: () => {
        return {
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartScreenCapture(captureParams) {
    return 'RtcEngine_startScreenCapture_270da41';
  }
  updateScreenCapture(captureParams) {
    const apiType = this.getApiTypeFromUpdateScreenCapture(captureParams);
    const jsonParams = {
      captureParams: captureParams,
      toJSON: () => {
        return {
          captureParams: captureParams
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateScreenCapture(captureParams) {
    return 'RtcEngine_updateScreenCapture_270da41';
  }
  queryScreenCaptureCapability() {
    const apiType = this.getApiTypeFromQueryScreenCaptureCapability();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromQueryScreenCaptureCapability() {
    return 'RtcEngine_queryScreenCaptureCapability';
  }
  queryCameraFocalLengthCapability() {
    const apiType = this.getApiTypeFromQueryCameraFocalLengthCapability();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const focalLengthInfos = jsonResults.focalLengthInfos;
    const size = jsonResults.size;
    return {
      focalLengthInfos,
      size
    };
  }
  getApiTypeFromQueryCameraFocalLengthCapability() {
    return 'RtcEngine_queryCameraFocalLengthCapability_2dee6af';
  }
  setExternalMediaProjection(mediaProjection) {
    const apiType = this.getApiTypeFromSetExternalMediaProjection(mediaProjection);
    const jsonParams = {
      mediaProjection: mediaProjection,
      toJSON: () => {
        return {
          mediaProjection: mediaProjection
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalMediaProjection(mediaProjection) {
    return 'RtcEngine_setExternalMediaProjection_f337cbf';
  }
  setScreenCaptureScenario(screenScenario) {
    const apiType = this.getApiTypeFromSetScreenCaptureScenario(screenScenario);
    const jsonParams = {
      screenScenario: screenScenario,
      toJSON: () => {
        return {
          screenScenario: screenScenario
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetScreenCaptureScenario(screenScenario) {
    return 'RtcEngine_setScreenCaptureScenario_13de7b4';
  }
  stopScreenCapture() {
    const apiType = this.getApiTypeFromStopScreenCapture();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopScreenCapture() {
    return 'RtcEngine_stopScreenCapture';
  }
  getCallId() {
    const apiType = this.getApiTypeFromGetCallId();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const callId = jsonResults.callId;
    return callId;
  }
  getApiTypeFromGetCallId() {
    return 'RtcEngine_getCallId_66d4ecd';
  }
  rate(callId, rating, description) {
    const apiType = this.getApiTypeFromRate(callId, rating, description);
    const jsonParams = {
      callId: callId,
      rating: rating,
      description: description,
      toJSON: () => {
        return {
          callId: callId,
          rating: rating,
          description: description
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRate(callId, rating, description) {
    return 'RtcEngine_rate_f1a0070';
  }
  complain(callId, description) {
    const apiType = this.getApiTypeFromComplain(callId, description);
    const jsonParams = {
      callId: callId,
      description: description,
      toJSON: () => {
        return {
          callId: callId,
          description: description
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromComplain(callId, description) {
    return 'RtcEngine_complain_ccad422';
  }
  startRtmpStreamWithoutTranscoding(url) {
    const apiType = this.getApiTypeFromStartRtmpStreamWithoutTranscoding(url);
    const jsonParams = {
      url: url,
      toJSON: () => {
        return {
          url: url
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRtmpStreamWithoutTranscoding(url) {
    return 'RtcEngine_startRtmpStreamWithoutTranscoding_3a2037f';
  }
  startRtmpStreamWithTranscoding(url, transcoding) {
    const apiType = this.getApiTypeFromStartRtmpStreamWithTranscoding(url, transcoding);
    const jsonParams = {
      url: url,
      transcoding: transcoding,
      toJSON: () => {
        return {
          url: url,
          transcoding: transcoding
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRtmpStreamWithTranscoding(url, transcoding) {
    return 'RtcEngine_startRtmpStreamWithTranscoding_f76aa1a';
  }
  updateRtmpTranscoding(transcoding) {
    const apiType = this.getApiTypeFromUpdateRtmpTranscoding(transcoding);
    const jsonParams = {
      transcoding: transcoding,
      toJSON: () => {
        return {
          transcoding: transcoding
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateRtmpTranscoding(transcoding) {
    return 'RtcEngine_updateRtmpTranscoding_91368d4';
  }
  startLocalVideoTranscoder(config) {
    const apiType = this.getApiTypeFromStartLocalVideoTranscoder(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartLocalVideoTranscoder(config) {
    return 'RtcEngine_startLocalVideoTranscoder_90f9e33';
  }
  updateLocalTranscoderConfiguration(config) {
    const apiType = this.getApiTypeFromUpdateLocalTranscoderConfiguration(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateLocalTranscoderConfiguration(config) {
    return 'RtcEngine_updateLocalTranscoderConfiguration_90f9e33';
  }
  stopRtmpStream(url) {
    const apiType = this.getApiTypeFromStopRtmpStream(url);
    const jsonParams = {
      url: url,
      toJSON: () => {
        return {
          url: url
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopRtmpStream(url) {
    return 'RtcEngine_stopRtmpStream_3a2037f';
  }
  stopLocalVideoTranscoder() {
    const apiType = this.getApiTypeFromStopLocalVideoTranscoder();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopLocalVideoTranscoder() {
    return 'RtcEngine_stopLocalVideoTranscoder';
  }
  startLocalAudioMixer(config) {
    const apiType = this.getApiTypeFromStartLocalAudioMixer(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartLocalAudioMixer(config) {
    return 'RtcEngine_startLocalAudioMixer_a7ff78e';
  }
  updateLocalAudioMixerConfiguration(config) {
    const apiType = this.getApiTypeFromUpdateLocalAudioMixerConfiguration(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateLocalAudioMixerConfiguration(config) {
    return 'RtcEngine_updateLocalAudioMixerConfiguration_a7ff78e';
  }
  stopLocalAudioMixer() {
    const apiType = this.getApiTypeFromStopLocalAudioMixer();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopLocalAudioMixer() {
    return 'RtcEngine_stopLocalAudioMixer';
  }
  startCameraCapture(sourceType, config) {
    const apiType = this.getApiTypeFromStartCameraCapture(sourceType, config);
    const jsonParams = {
      sourceType: sourceType,
      config: config,
      toJSON: () => {
        return {
          sourceType: sourceType,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartCameraCapture(sourceType, config) {
    return 'RtcEngine_startCameraCapture_f3692cc';
  }
  stopCameraCapture(sourceType) {
    const apiType = this.getApiTypeFromStopCameraCapture(sourceType);
    const jsonParams = {
      sourceType: sourceType,
      toJSON: () => {
        return {
          sourceType: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopCameraCapture(sourceType) {
    return 'RtcEngine_stopCameraCapture_4fd718e';
  }
  setCameraDeviceOrientation(type, orientation) {
    const apiType = this.getApiTypeFromSetCameraDeviceOrientation(type, orientation);
    const jsonParams = {
      type: type,
      orientation: orientation,
      toJSON: () => {
        return {
          type: type,
          orientation: orientation
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCameraDeviceOrientation(type, orientation) {
    return 'RtcEngine_setCameraDeviceOrientation_025aae8';
  }
  setScreenCaptureOrientation(type, orientation) {
    const apiType = this.getApiTypeFromSetScreenCaptureOrientation(type, orientation);
    const jsonParams = {
      type: type,
      orientation: orientation,
      toJSON: () => {
        return {
          type: type,
          orientation: orientation
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetScreenCaptureOrientation(type, orientation) {
    return 'RtcEngine_setScreenCaptureOrientation_025aae8';
  }
  getConnectionState() {
    const apiType = this.getApiTypeFromGetConnectionState();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetConnectionState() {
    return 'RtcEngine_getConnectionState';
  }
  registerEventHandler(eventHandler) {
    const apiType = this.getApiTypeFromRegisterEventHandler(eventHandler);
    const jsonParams = {
      eventHandler: eventHandler,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterEventHandler(eventHandler) {
    return 'RtcEngine_registerEventHandler_5fc0465';
  }
  unregisterEventHandler(eventHandler) {
    const apiType = this.getApiTypeFromUnregisterEventHandler(eventHandler);
    const jsonParams = {
      eventHandler: eventHandler,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterEventHandler(eventHandler) {
    return 'RtcEngine_unregisterEventHandler_5fc0465';
  }
  setRemoteUserPriority(uid, userPriority) {
    const apiType = this.getApiTypeFromSetRemoteUserPriority(uid, userPriority);
    const jsonParams = {
      uid: uid,
      userPriority: userPriority,
      toJSON: () => {
        return {
          uid: uid,
          userPriority: userPriority
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteUserPriority(uid, userPriority) {
    return 'RtcEngine_setRemoteUserPriority_f34115b';
  }
  enableEncryption(enabled, config) {
    const apiType = this.getApiTypeFromEnableEncryption(enabled, config);
    const jsonParams = {
      enabled: enabled,
      config: config,
      toJSON: () => {
        return {
          enabled: enabled,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableEncryption(enabled, config) {
    return 'RtcEngine_enableEncryption_421c27b';
  }
  createDataStream(config) {
    const apiType = this.getApiTypeFromCreateDataStream(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const streamId = jsonResults.streamId;
    return streamId;
  }
  getApiTypeFromCreateDataStream(config) {
    return 'RtcEngine_createDataStream_5862815';
  }
  sendStreamMessage(streamId, data, length) {
    const apiType = this.getApiTypeFromSendStreamMessage(streamId, data, length);
    const jsonParams = {
      streamId: streamId,
      data: data,
      length: length,
      toJSON: () => {
        return {
          streamId: streamId,
          length: length
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendStreamMessage(streamId, data, length) {
    return 'RtcEngine_sendStreamMessage_8715a45';
  }
  addVideoWatermark(watermarkUrl, options) {
    const apiType = this.getApiTypeFromAddVideoWatermark(watermarkUrl, options);
    const jsonParams = {
      watermarkUrl: watermarkUrl,
      options: options,
      toJSON: () => {
        return {
          watermarkUrl: watermarkUrl,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAddVideoWatermark(watermarkUrl, options) {
    return 'RtcEngine_addVideoWatermark_7480410';
  }
  clearVideoWatermarks() {
    const apiType = this.getApiTypeFromClearVideoWatermarks();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromClearVideoWatermarks() {
    return 'RtcEngine_clearVideoWatermarks';
  }
  pauseAudio() {
    const apiType = this.getApiTypeFromPauseAudio();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseAudio() {
    return 'RtcEngine_pauseAudio';
  }
  resumeAudio() {
    const apiType = this.getApiTypeFromResumeAudio();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeAudio() {
    return 'RtcEngine_resumeAudio';
  }
  enableWebSdkInteroperability(enabled) {
    const apiType = this.getApiTypeFromEnableWebSdkInteroperability(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableWebSdkInteroperability(enabled) {
    return 'RtcEngine_enableWebSdkInteroperability_5039d15';
  }
  sendCustomReportMessage(id, category, event, label, value) {
    const apiType = this.getApiTypeFromSendCustomReportMessage(id, category, event, label, value);
    const jsonParams = {
      id: id,
      category: category,
      event: event,
      label: label,
      value: value,
      toJSON: () => {
        return {
          id: id,
          category: category,
          event: event,
          label: label,
          value: value
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendCustomReportMessage(id, category, event, label, value) {
    return 'RtcEngine_sendCustomReportMessage_56d6589';
  }
  registerMediaMetadataObserver(observer, type) {
    const apiType = this.getApiTypeFromRegisterMediaMetadataObserver(observer, type);
    const jsonParams = {
      observer: observer,
      type: type,
      toJSON: () => {
        return {
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterMediaMetadataObserver(observer, type) {
    return 'RtcEngine_registerMediaMetadataObserver_8701fec';
  }
  unregisterMediaMetadataObserver(observer, type) {
    const apiType = this.getApiTypeFromUnregisterMediaMetadataObserver(observer, type);
    const jsonParams = {
      observer: observer,
      type: type,
      toJSON: () => {
        return {
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterMediaMetadataObserver(observer, type) {
    return 'RtcEngine_unregisterMediaMetadataObserver_8701fec';
  }
  startAudioFrameDump(channelId, uid, location, uuid, passwd, durationMs, autoUpload) {
    const apiType = this.getApiTypeFromStartAudioFrameDump(channelId, uid, location, uuid, passwd, durationMs, autoUpload);
    const jsonParams = {
      channel_id: channelId,
      uid: uid,
      location: location,
      uuid: uuid,
      passwd: passwd,
      duration_ms: durationMs,
      auto_upload: autoUpload,
      toJSON: () => {
        return {
          channel_id: channelId,
          uid: uid,
          location: location,
          uuid: uuid,
          passwd: passwd,
          duration_ms: durationMs,
          auto_upload: autoUpload
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartAudioFrameDump(channelId, uid, location, uuid, passwd, durationMs, autoUpload) {
    return 'RtcEngine_startAudioFrameDump_aad7331';
  }
  stopAudioFrameDump(channelId, uid, location) {
    const apiType = this.getApiTypeFromStopAudioFrameDump(channelId, uid, location);
    const jsonParams = {
      channel_id: channelId,
      uid: uid,
      location: location,
      toJSON: () => {
        return {
          channel_id: channelId,
          uid: uid,
          location: location
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopAudioFrameDump(channelId, uid, location) {
    return 'RtcEngine_stopAudioFrameDump_a4c9af4';
  }
  setAINSMode(enabled, mode) {
    const apiType = this.getApiTypeFromSetAINSMode(enabled, mode);
    const jsonParams = {
      enabled: enabled,
      mode: mode,
      toJSON: () => {
        return {
          enabled: enabled,
          mode: mode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAINSMode(enabled, mode) {
    return 'RtcEngine_setAINSMode_4df3049';
  }
  registerLocalUserAccount(appId, userAccount) {
    const apiType = this.getApiTypeFromRegisterLocalUserAccount(appId, userAccount);
    const jsonParams = {
      appId: appId,
      userAccount: userAccount,
      toJSON: () => {
        return {
          appId: appId,
          userAccount: userAccount
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterLocalUserAccount(appId, userAccount) {
    return 'RtcEngine_registerLocalUserAccount_ccad422';
  }
  joinChannelWithUserAccount(token, channelId, userAccount, options) {
    const apiType = this.getApiTypeFromJoinChannelWithUserAccount(token, channelId, userAccount, options);
    const jsonParams = {
      token: token,
      channelId: channelId,
      userAccount: userAccount,
      options: options,
      toJSON: () => {
        return {
          token: token,
          channelId: channelId,
          userAccount: userAccount,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromJoinChannelWithUserAccount(token, channelId, userAccount, options) {
    return 'RtcEngine_joinChannelWithUserAccount_4685af9';
  }
  joinChannelWithUserAccountEx(token, channelId, userAccount, options) {
    const apiType = this.getApiTypeFromJoinChannelWithUserAccountEx(token, channelId, userAccount, options);
    const jsonParams = {
      token: token,
      channelId: channelId,
      userAccount: userAccount,
      options: options,
      toJSON: () => {
        return {
          token: token,
          channelId: channelId,
          userAccount: userAccount,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromJoinChannelWithUserAccountEx(token, channelId, userAccount, options) {
    return 'RtcEngine_joinChannelWithUserAccountEx_268b977';
  }
  getUserInfoByUserAccount(userAccount) {
    const apiType = this.getApiTypeFromGetUserInfoByUserAccount(userAccount);
    const jsonParams = {
      userAccount: userAccount,
      toJSON: () => {
        return {
          userAccount: userAccount
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const userInfo = jsonResults.userInfo;
    return userInfo;
  }
  getApiTypeFromGetUserInfoByUserAccount(userAccount) {
    return 'RtcEngine_getUserInfoByUserAccount_c6a8f08';
  }
  getUserInfoByUid(uid) {
    const apiType = this.getApiTypeFromGetUserInfoByUid(uid);
    const jsonParams = {
      uid: uid,
      toJSON: () => {
        return {
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const userInfo = jsonResults.userInfo;
    return userInfo;
  }
  getApiTypeFromGetUserInfoByUid(uid) {
    return 'RtcEngine_getUserInfoByUid_6b7aee8';
  }
  startOrUpdateChannelMediaRelay(configuration) {
    const apiType = this.getApiTypeFromStartOrUpdateChannelMediaRelay(configuration);
    const jsonParams = {
      configuration: configuration,
      toJSON: () => {
        return {
          configuration: configuration
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartOrUpdateChannelMediaRelay(configuration) {
    return 'RtcEngine_startOrUpdateChannelMediaRelay_e68f0a4';
  }
  stopChannelMediaRelay() {
    const apiType = this.getApiTypeFromStopChannelMediaRelay();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopChannelMediaRelay() {
    return 'RtcEngine_stopChannelMediaRelay';
  }
  pauseAllChannelMediaRelay() {
    const apiType = this.getApiTypeFromPauseAllChannelMediaRelay();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseAllChannelMediaRelay() {
    return 'RtcEngine_pauseAllChannelMediaRelay';
  }
  resumeAllChannelMediaRelay() {
    const apiType = this.getApiTypeFromResumeAllChannelMediaRelay();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeAllChannelMediaRelay() {
    return 'RtcEngine_resumeAllChannelMediaRelay';
  }
  setDirectCdnStreamingAudioConfiguration(profile) {
    const apiType = this.getApiTypeFromSetDirectCdnStreamingAudioConfiguration(profile);
    const jsonParams = {
      profile: profile,
      toJSON: () => {
        return {
          profile: profile
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDirectCdnStreamingAudioConfiguration(profile) {
    return 'RtcEngine_setDirectCdnStreamingAudioConfiguration_ac39c15';
  }
  setDirectCdnStreamingVideoConfiguration(config) {
    const apiType = this.getApiTypeFromSetDirectCdnStreamingVideoConfiguration(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDirectCdnStreamingVideoConfiguration(config) {
    return 'RtcEngine_setDirectCdnStreamingVideoConfiguration_89677d8';
  }
  startDirectCdnStreaming(eventHandler, publishUrl, options) {
    const apiType = this.getApiTypeFromStartDirectCdnStreaming(eventHandler, publishUrl, options);
    const jsonParams = {
      eventHandler: eventHandler,
      publishUrl: publishUrl,
      options: options,
      toJSON: () => {
        return {
          publishUrl: publishUrl,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartDirectCdnStreaming(eventHandler, publishUrl, options) {
    return 'RtcEngine_startDirectCdnStreaming_ed8d77b';
  }
  stopDirectCdnStreaming() {
    const apiType = this.getApiTypeFromStopDirectCdnStreaming();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopDirectCdnStreaming() {
    return 'RtcEngine_stopDirectCdnStreaming';
  }
  updateDirectCdnStreamingMediaOptions(options) {
    const apiType = this.getApiTypeFromUpdateDirectCdnStreamingMediaOptions(options);
    const jsonParams = {
      options: options,
      toJSON: () => {
        return {
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateDirectCdnStreamingMediaOptions(options) {
    return 'RtcEngine_updateDirectCdnStreamingMediaOptions_d2556c8';
  }
  startRhythmPlayer(sound1, sound2, config) {
    const apiType = this.getApiTypeFromStartRhythmPlayer(sound1, sound2, config);
    const jsonParams = {
      sound1: sound1,
      sound2: sound2,
      config: config,
      toJSON: () => {
        return {
          sound1: sound1,
          sound2: sound2,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRhythmPlayer(sound1, sound2, config) {
    return 'RtcEngine_startRhythmPlayer_e1f6565';
  }
  stopRhythmPlayer() {
    const apiType = this.getApiTypeFromStopRhythmPlayer();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopRhythmPlayer() {
    return 'RtcEngine_stopRhythmPlayer';
  }
  configRhythmPlayer(config) {
    const apiType = this.getApiTypeFromConfigRhythmPlayer(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromConfigRhythmPlayer(config) {
    return 'RtcEngine_configRhythmPlayer_b36c805';
  }
  takeSnapshot(uid, filePath) {
    const apiType = this.getApiTypeFromTakeSnapshot(uid, filePath);
    const jsonParams = {
      uid: uid,
      filePath: filePath,
      toJSON: () => {
        return {
          uid: uid,
          filePath: filePath
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromTakeSnapshot(uid, filePath) {
    return 'RtcEngine_takeSnapshot_1922dd1';
  }
  enableContentInspect(enabled, config) {
    const apiType = this.getApiTypeFromEnableContentInspect(enabled, config);
    const jsonParams = {
      enabled: enabled,
      config: config,
      toJSON: () => {
        return {
          enabled: enabled,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableContentInspect(enabled, config) {
    return 'RtcEngine_enableContentInspect_e15e514';
  }
  adjustCustomAudioPublishVolume(trackId, volume) {
    const apiType = this.getApiTypeFromAdjustCustomAudioPublishVolume(trackId, volume);
    const jsonParams = {
      trackId: trackId,
      volume: volume,
      toJSON: () => {
        return {
          trackId: trackId,
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustCustomAudioPublishVolume(trackId, volume) {
    return 'RtcEngine_adjustCustomAudioPublishVolume_f8da2ca';
  }
  adjustCustomAudioPlayoutVolume(trackId, volume) {
    const apiType = this.getApiTypeFromAdjustCustomAudioPlayoutVolume(trackId, volume);
    const jsonParams = {
      trackId: trackId,
      volume: volume,
      toJSON: () => {
        return {
          trackId: trackId,
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustCustomAudioPlayoutVolume(trackId, volume) {
    return 'RtcEngine_adjustCustomAudioPlayoutVolume_f8da2ca';
  }
  setCloudProxy(proxyType) {
    const apiType = this.getApiTypeFromSetCloudProxy(proxyType);
    const jsonParams = {
      proxyType: proxyType,
      toJSON: () => {
        return {
          proxyType: proxyType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCloudProxy(proxyType) {
    return 'RtcEngine_setCloudProxy_39d115e';
  }
  setLocalAccessPoint(config) {
    const apiType = this.getApiTypeFromSetLocalAccessPoint(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLocalAccessPoint(config) {
    return 'RtcEngine_setLocalAccessPoint_798c8c7';
  }
  setAdvancedAudioOptions(options) {
    let sourceType = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    const apiType = this.getApiTypeFromSetAdvancedAudioOptions(options, sourceType);
    const jsonParams = {
      options: options,
      sourceType: sourceType,
      toJSON: () => {
        return {
          options: options,
          sourceType: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAdvancedAudioOptions(options) {
    let sourceType = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'RtcEngine_setAdvancedAudioOptions_38d986b';
  }
  setAVSyncSource(channelId, uid) {
    const apiType = this.getApiTypeFromSetAVSyncSource(channelId, uid);
    const jsonParams = {
      channelId: channelId,
      uid: uid,
      toJSON: () => {
        return {
          channelId: channelId,
          uid: uid
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAVSyncSource(channelId, uid) {
    return 'RtcEngine_setAVSyncSource_bf26e54';
  }
  enableVideoImageSource(enable, options) {
    const apiType = this.getApiTypeFromEnableVideoImageSource(enable, options);
    const jsonParams = {
      enable: enable,
      options: options,
      toJSON: () => {
        return {
          enable: enable,
          options: options
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableVideoImageSource(enable, options) {
    return 'RtcEngine_enableVideoImageSource_5f39ea0';
  }
  getCurrentMonotonicTimeInMs() {
    const apiType = this.getApiTypeFromGetCurrentMonotonicTimeInMs();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCurrentMonotonicTimeInMs() {
    return 'RtcEngine_getCurrentMonotonicTimeInMs';
  }
  enableWirelessAccelerate(enabled) {
    const apiType = this.getApiTypeFromEnableWirelessAccelerate(enabled);
    const jsonParams = {
      enabled: enabled,
      toJSON: () => {
        return {
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableWirelessAccelerate(enabled) {
    return 'RtcEngine_enableWirelessAccelerate_5039d15';
  }
  getNetworkType() {
    const apiType = this.getApiTypeFromGetNetworkType();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetNetworkType() {
    return 'RtcEngine_getNetworkType';
  }
  setParameters(parameters) {
    const apiType = this.getApiTypeFromSetParameters(parameters);
    const jsonParams = {
      parameters: parameters,
      toJSON: () => {
        return {
          parameters: parameters
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetParameters(parameters) {
    return 'RtcEngine_setParameters_3a2037f';
  }
  startMediaRenderingTracing() {
    const apiType = this.getApiTypeFromStartMediaRenderingTracing();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartMediaRenderingTracing() {
    return 'RtcEngine_startMediaRenderingTracing';
  }
  enableInstantMediaRendering() {
    const apiType = this.getApiTypeFromEnableInstantMediaRendering();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableInstantMediaRendering() {
    return 'RtcEngine_enableInstantMediaRendering';
  }
  getNtpWallTimeInMs() {
    const apiType = this.getApiTypeFromGetNtpWallTimeInMs();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetNtpWallTimeInMs() {
    return 'RtcEngine_getNtpWallTimeInMs';
  }
  isFeatureAvailableOnDevice(type) {
    const apiType = this.getApiTypeFromIsFeatureAvailableOnDevice(type);
    const jsonParams = {
      type: type,
      toJSON: () => {
        return {
          type: type
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromIsFeatureAvailableOnDevice(type) {
    return 'RtcEngine_isFeatureAvailableOnDevice_a694b62';
  }
  sendAudioMetadata(metadata, length) {
    const apiType = this.getApiTypeFromSendAudioMetadata(metadata, length);
    const jsonParams = {
      metadata: metadata,
      length: length,
      toJSON: () => {
        return {
          metadata: metadata,
          length: length
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendAudioMetadata(metadata, length) {
    return 'RtcEngine_sendAudioMetadata_878f309';
  }
  queryHDRCapability(videoModule) {
    const apiType = this.getApiTypeFromQueryHDRCapability(videoModule);
    const jsonParams = {
      videoModule: videoModule,
      toJSON: () => {
        return {
          videoModule: videoModule
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const capability = jsonResults.capability;
    return capability;
  }
  getApiTypeFromQueryHDRCapability(videoModule) {
    return 'RtcEngine_queryHDRCapability_bebdacb';
  }
  startScreenCaptureBySourceType(sourceType, config) {
    const apiType = this.getApiTypeFromStartScreenCaptureBySourceType(sourceType, config);
    const jsonParams = {
      sourceType: sourceType,
      config: config,
      toJSON: () => {
        return {
          sourceType: sourceType,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartScreenCaptureBySourceType(sourceType, config) {
    return 'RtcEngine_startScreenCapture_9ebb320';
  }
  stopScreenCaptureBySourceType(sourceType) {
    const apiType = this.getApiTypeFromStopScreenCaptureBySourceType(sourceType);
    const jsonParams = {
      sourceType: sourceType,
      toJSON: () => {
        return {
          sourceType: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopScreenCaptureBySourceType(sourceType) {
    return 'RtcEngine_stopScreenCapture_4fd718e';
  }
  release() {
    let sync = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    const apiType = this.getApiTypeFromRelease(sync);
    const jsonParams = {
      sync: sync,
      toJSON: () => {
        return {
          sync: sync
        };
      }
    };
    _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromRelease() {
    let sync = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    return 'RtcEngine_release';
  }
  startPreviewWithoutSourceType() {
    const apiType = this.getApiTypeFromStartPreviewWithoutSourceType();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartPreviewWithoutSourceType() {
    return 'RtcEngine_startPreview';
  }
  getAudioDeviceManager() {
    const apiType = this.getApiTypeFromGetAudioDeviceManager();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAudioDeviceManager() {
    return 'RtcEngine_getAudioDeviceManager';
  }
  getVideoDeviceManager() {
    const apiType = this.getApiTypeFromGetVideoDeviceManager();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetVideoDeviceManager() {
    return 'RtcEngine_getVideoDeviceManager';
  }
  getMusicContentCenter() {
    const apiType = this.getApiTypeFromGetMusicContentCenter();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMusicContentCenter() {
    return 'RtcEngine_getMusicContentCenter';
  }
  getMediaEngine() {
    const apiType = this.getApiTypeFromGetMediaEngine();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMediaEngine() {
    return 'RtcEngine_getMediaEngine';
  }
  getLocalSpatialAudioEngine() {
    const apiType = this.getApiTypeFromGetLocalSpatialAudioEngine();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetLocalSpatialAudioEngine() {
    return 'RtcEngine_getLocalSpatialAudioEngine';
  }
  getH265Transcoder() {
    const apiType = this.getApiTypeFromGetH265Transcoder();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetH265Transcoder() {
    return 'RtcEngine_getH265Transcoder';
  }
  sendMetaData(metadata, sourceType) {
    const apiType = this.getApiTypeFromSendMetaData(metadata, sourceType);
    const jsonParams = {
      metadata: metadata,
      source_type: sourceType,
      toJSON: () => {
        return {
          metadata: metadata,
          source_type: sourceType
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendMetaData(metadata, sourceType) {
    return 'RtcEngine_sendMetaData';
  }
  setMaxMetadataSize(size) {
    const apiType = this.getApiTypeFromSetMaxMetadataSize(size);
    const jsonParams = {
      size: size,
      toJSON: () => {
        return {
          size: size
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetMaxMetadataSize(size) {
    return 'RtcEngine_setMaxMetadataSize';
  }
  destroyRendererByView(view) {
    const apiType = this.getApiTypeFromDestroyRendererByView(view);
    const jsonParams = {
      view: view,
      toJSON: () => {
        return {
          view: view
        };
      }
    };
    _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromDestroyRendererByView(view) {
    return 'RtcEngine_destroyRendererByView_a55f55f';
  }
  destroyRendererByConfig(sourceType, channelId) {
    let uid = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
    const apiType = this.getApiTypeFromDestroyRendererByConfig(sourceType, channelId, uid);
    const jsonParams = {
      sourceType: sourceType,
      channelId: channelId,
      uid: uid,
      toJSON: () => {
        return {
          sourceType: sourceType,
          channelId: channelId,
          uid: uid
        };
      }
    };
    _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromDestroyRendererByConfig(sourceType, channelId) {
    let uid = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
    return 'RtcEngine_destroyRendererByConfig_542c2ae';
  }
  unregisterAudioEncodedFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterAudioEncodedFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterAudioEncodedFrameObserver(observer) {
    return 'RtcEngine_unregisterAudioEncodedFrameObserver';
  }
  getNativeHandle() {
    const apiType = this.getApiTypeFromGetNativeHandle();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetNativeHandle() {
    return 'RtcEngine_getNativeHandle';
  }
  takeSnapshotWithConfig(uid, config) {
    const apiType = this.getApiTypeFromTakeSnapshotWithConfig(uid, config);
    const jsonParams = {
      uid: uid,
      config: config,
      toJSON: () => {
        return {
          uid: uid,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromTakeSnapshotWithConfig(uid, config) {
    return 'RtcEngine_takeSnapshot_5669ea6';
  }
}
exports.IRtcEngineImpl = IRtcEngineImpl;
