// react-native-agora/lib/commonjs/impl/IAgoraMediaEngineImpl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IMediaEngineImpl = void 0;
var _AgoraBase = require("../AgoraBase");
var _AgoraMediaBase = require("../AgoraMediaBase");
var _IrisApiEngine = require("../internal/IrisApiEngine");
// @ts-ignore
class IMediaEngineImpl {
  registerAudioFrameObserver(observer) {
    const apiType = this.getApiTypeFromRegisterAudioFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterAudioFrameObserver(observer) {
    return 'MediaEngine_registerAudioFrameObserver_d873a64';
  }
  registerVideoFrameObserver(observer) {
    const apiType = this.getApiTypeFromRegisterVideoFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterVideoFrameObserver(observer) {
    return 'MediaEngine_registerVideoFrameObserver_2cc0ef1';
  }
  registerVideoEncodedFrameObserver(observer) {
    const apiType = this.getApiTypeFromRegisterVideoEncodedFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterVideoEncodedFrameObserver(observer) {
    return 'MediaEngine_registerVideoEncodedFrameObserver_d45d579';
  }
  registerFaceInfoObserver(observer) {
    const apiType = this.getApiTypeFromRegisterFaceInfoObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterFaceInfoObserver(observer) {
    return 'MediaEngine_registerFaceInfoObserver_0303ed6';
  }
  pushAudioFrame(frame) {
    let trackId = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    const apiType = this.getApiTypeFromPushAudioFrame(frame, trackId);
    const jsonParams = {
      frame: frame,
      trackId: trackId,
      toJSON: () => {
        return {
          frame: frame,
          trackId: trackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPushAudioFrame(frame) {
    let trackId = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'MediaEngine_pushAudioFrame_c71f4ab';
  }
  pullAudioFrame(frame) {
    const apiType = this.getApiTypeFromPullAudioFrame(frame);
    const jsonParams = {
      frame: frame,
      toJSON: () => {
        return {
          frame: frame
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPullAudioFrame(frame) {
    return 'MediaEngine_pullAudioFrame_2c74a9c';
  }
  setExternalVideoSource(enabled, useTexture) {
    let sourceType = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.ExternalVideoSourceType.VideoFrame;
    let encodedVideoOption = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : new _AgoraBase.SenderOptions();
    const apiType = this.getApiTypeFromSetExternalVideoSource(enabled, useTexture, sourceType, encodedVideoOption);
    const jsonParams = {
      enabled: enabled,
      useTexture: useTexture,
      sourceType: sourceType,
      encodedVideoOption: encodedVideoOption,
      toJSON: () => {
        return {
          enabled: enabled,
          useTexture: useTexture,
          sourceType: sourceType,
          encodedVideoOption: encodedVideoOption
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalVideoSource(enabled, useTexture) {
    let sourceType = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _AgoraMediaBase.ExternalVideoSourceType.VideoFrame;
    let encodedVideoOption = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : new _AgoraBase.SenderOptions();
    return 'MediaEngine_setExternalVideoSource_fff99b6';
  }
  setExternalRemoteEglContext(eglContext) {
    const apiType = this.getApiTypeFromSetExternalRemoteEglContext(eglContext);
    const jsonParams = {
      eglContext: eglContext,
      toJSON: () => {
        return {
          eglContext: eglContext
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalRemoteEglContext(eglContext) {
    return 'MediaEngine_setExternalRemoteEglContext_f337cbf';
  }
  setExternalAudioSource(enabled, sampleRate, channels) {
    let localPlayback = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;
    let publish = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : true;
    const apiType = this.getApiTypeFromSetExternalAudioSource(enabled, sampleRate, channels, localPlayback, publish);
    const jsonParams = {
      enabled: enabled,
      sampleRate: sampleRate,
      channels: channels,
      localPlayback: localPlayback,
      publish: publish,
      toJSON: () => {
        return {
          enabled: enabled,
          sampleRate: sampleRate,
          channels: channels,
          localPlayback: localPlayback,
          publish: publish
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalAudioSource(enabled, sampleRate, channels) {
    let localPlayback = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;
    let publish = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : true;
    return 'MediaEngine_setExternalAudioSource_e6538be';
  }
  createCustomAudioTrack(trackType, config) {
    const apiType = this.getApiTypeFromCreateCustomAudioTrack(trackType, config);
    const jsonParams = {
      trackType: trackType,
      config: config,
      toJSON: () => {
        return {
          trackType: trackType,
          config: config
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromCreateCustomAudioTrack(trackType, config) {
    return 'MediaEngine_createCustomAudioTrack_5a0bf1a';
  }
  destroyCustomAudioTrack(trackId) {
    const apiType = this.getApiTypeFromDestroyCustomAudioTrack(trackId);
    const jsonParams = {
      trackId: trackId,
      toJSON: () => {
        return {
          trackId: trackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromDestroyCustomAudioTrack(trackId) {
    return 'MediaEngine_destroyCustomAudioTrack_6178b5d';
  }
  setExternalAudioSink(enabled, sampleRate, channels) {
    const apiType = this.getApiTypeFromSetExternalAudioSink(enabled, sampleRate, channels);
    const jsonParams = {
      enabled: enabled,
      sampleRate: sampleRate,
      channels: channels,
      toJSON: () => {
        return {
          enabled: enabled,
          sampleRate: sampleRate,
          channels: channels
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalAudioSink(enabled, sampleRate, channels) {
    return 'MediaEngine_setExternalAudioSink_d275ce0';
  }
  enableCustomAudioLocalPlayback(trackId, enabled) {
    const apiType = this.getApiTypeFromEnableCustomAudioLocalPlayback(trackId, enabled);
    const jsonParams = {
      trackId: trackId,
      enabled: enabled,
      toJSON: () => {
        return {
          trackId: trackId,
          enabled: enabled
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableCustomAudioLocalPlayback(trackId, enabled) {
    return 'MediaEngine_enableCustomAudioLocalPlayback_5f38e8a';
  }
  pushVideoFrame(frame) {
    let videoTrackId = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    const apiType = this.getApiTypeFromPushVideoFrame(frame, videoTrackId);
    const jsonParams = {
      frame: frame,
      videoTrackId: videoTrackId,
      toJSON: () => {
        return {
          frame: frame,
          videoTrackId: videoTrackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPushVideoFrame(frame) {
    let videoTrackId = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'MediaEngine_pushVideoFrame_4e544e2';
  }
  pushEncodedVideoImage(imageBuffer, length, videoEncodedFrameInfo) {
    let videoTrackId = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 0;
    const apiType = this.getApiTypeFromPushEncodedVideoImage(imageBuffer, length, videoEncodedFrameInfo, videoTrackId);
    const jsonParams = {
      imageBuffer: imageBuffer,
      length: length,
      videoEncodedFrameInfo: videoEncodedFrameInfo,
      videoTrackId: videoTrackId,
      toJSON: () => {
        return {
          length: length,
          videoEncodedFrameInfo: videoEncodedFrameInfo,
          videoTrackId: videoTrackId
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPushEncodedVideoImage(imageBuffer, length, videoEncodedFrameInfo) {
    let videoTrackId = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 0;
    return 'MediaEngine_pushEncodedVideoImage_e71452b';
  }
  release() {
    const apiType = this.getApiTypeFromRelease();
    const jsonParams = {};
    _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
  }
  getApiTypeFromRelease() {
    return 'MediaEngine_release';
  }
  unregisterAudioFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterAudioFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterAudioFrameObserver(observer) {
    return 'MediaEngine_unregisterAudioFrameObserver';
  }
  unregisterVideoFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterVideoFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterVideoFrameObserver(observer) {
    return 'MediaEngine_unregisterVideoFrameObserver';
  }
  unregisterVideoEncodedFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterVideoEncodedFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterVideoEncodedFrameObserver(observer) {
    return 'MediaEngine_unregisterVideoEncodedFrameObserver';
  }
  unregisterFaceInfoObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterFaceInfoObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterFaceInfoObserver(observer) {
    return 'MediaEngine_unregisterFaceInfoObserver';
  }
}
exports.IMediaEngineImpl = IMediaEngineImpl;
